{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Making a jupyter notebook for the processing of sequence files in Fall 2018.\n",
    "\n",
    "## Prepare the directories for sequence storage"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/json": {
       "cell": {
        "!": "OSMagics",
        "HTML": "Other",
        "SVG": "Other",
        "bash": "Other",
        "capture": "ExecutionMagics",
        "debug": "ExecutionMagics",
        "file": "Other",
        "html": "DisplayMagics",
        "javascript": "DisplayMagics",
        "js": "DisplayMagics",
        "latex": "DisplayMagics",
        "markdown": "DisplayMagics",
        "perl": "Other",
        "prun": "ExecutionMagics",
        "pypy": "Other",
        "python": "Other",
        "python2": "Other",
        "python3": "Other",
        "ruby": "Other",
        "script": "ScriptMagics",
        "sh": "Other",
        "svg": "DisplayMagics",
        "sx": "OSMagics",
        "system": "OSMagics",
        "time": "ExecutionMagics",
        "timeit": "ExecutionMagics",
        "writefile": "OSMagics"
       },
       "line": {
        "alias": "OSMagics",
        "alias_magic": "BasicMagics",
        "autocall": "AutoMagics",
        "automagic": "AutoMagics",
        "autosave": "KernelMagics",
        "bookmark": "OSMagics",
        "cat": "Other",
        "cd": "OSMagics",
        "clear": "KernelMagics",
        "colors": "BasicMagics",
        "config": "ConfigMagics",
        "connect_info": "KernelMagics",
        "cp": "Other",
        "debug": "ExecutionMagics",
        "dhist": "OSMagics",
        "dirs": "OSMagics",
        "doctest_mode": "BasicMagics",
        "ed": "Other",
        "edit": "KernelMagics",
        "env": "OSMagics",
        "gui": "BasicMagics",
        "hist": "Other",
        "history": "HistoryMagics",
        "killbgscripts": "ScriptMagics",
        "ldir": "Other",
        "less": "KernelMagics",
        "lf": "Other",
        "lk": "Other",
        "ll": "Other",
        "load": "CodeMagics",
        "load_ext": "ExtensionMagics",
        "loadpy": "CodeMagics",
        "logoff": "LoggingMagics",
        "logon": "LoggingMagics",
        "logstart": "LoggingMagics",
        "logstate": "LoggingMagics",
        "logstop": "LoggingMagics",
        "ls": "Other",
        "lsmagic": "BasicMagics",
        "lx": "Other",
        "macro": "ExecutionMagics",
        "magic": "BasicMagics",
        "man": "KernelMagics",
        "matplotlib": "PylabMagics",
        "mkdir": "Other",
        "more": "KernelMagics",
        "mv": "Other",
        "notebook": "BasicMagics",
        "page": "BasicMagics",
        "pastebin": "CodeMagics",
        "pdb": "ExecutionMagics",
        "pdef": "NamespaceMagics",
        "pdoc": "NamespaceMagics",
        "pfile": "NamespaceMagics",
        "pinfo": "NamespaceMagics",
        "pinfo2": "NamespaceMagics",
        "pip": "BasicMagics",
        "popd": "OSMagics",
        "pprint": "BasicMagics",
        "precision": "BasicMagics",
        "profile": "BasicMagics",
        "prun": "ExecutionMagics",
        "psearch": "NamespaceMagics",
        "psource": "NamespaceMagics",
        "pushd": "OSMagics",
        "pwd": "OSMagics",
        "pycat": "OSMagics",
        "pylab": "PylabMagics",
        "qtconsole": "KernelMagics",
        "quickref": "BasicMagics",
        "recall": "HistoryMagics",
        "rehashx": "OSMagics",
        "reload_ext": "ExtensionMagics",
        "rep": "Other",
        "rerun": "HistoryMagics",
        "reset": "NamespaceMagics",
        "reset_selective": "NamespaceMagics",
        "rm": "Other",
        "rmdir": "Other",
        "run": "ExecutionMagics",
        "save": "CodeMagics",
        "sc": "OSMagics",
        "set_env": "OSMagics",
        "store": "StoreMagics",
        "sx": "OSMagics",
        "system": "OSMagics",
        "tb": "ExecutionMagics",
        "time": "ExecutionMagics",
        "timeit": "ExecutionMagics",
        "unalias": "OSMagics",
        "unload_ext": "ExtensionMagics",
        "who": "NamespaceMagics",
        "who_ls": "NamespaceMagics",
        "whos": "NamespaceMagics",
        "xdel": "NamespaceMagics",
        "xmode": "BasicMagics"
       }
      },
      "text/plain": [
       "Available line magics:\n",
       "%alias  %alias_magic  %autocall  %automagic  %autosave  %bookmark  %cat  %cd  %clear  %colors  %config  %connect_info  %cp  %debug  %dhist  %dirs  %doctest_mode  %ed  %edit  %env  %gui  %hist  %history  %killbgscripts  %ldir  %less  %lf  %lk  %ll  %load  %load_ext  %loadpy  %logoff  %logon  %logstart  %logstate  %logstop  %ls  %lsmagic  %lx  %macro  %magic  %man  %matplotlib  %mkdir  %more  %mv  %notebook  %page  %pastebin  %pdb  %pdef  %pdoc  %pfile  %pinfo  %pinfo2  %popd  %pprint  %precision  %profile  %prun  %psearch  %psource  %pushd  %pwd  %pycat  %pylab  %qtconsole  %quickref  %recall  %rehashx  %reload_ext  %rep  %rerun  %reset  %reset_selective  %rm  %rmdir  %run  %save  %sc  %set_env  %store  %sx  %system  %tb  %time  %timeit  %unalias  %unload_ext  %who  %who_ls  %whos  %xdel  %xmode\n",
       "\n",
       "Available cell magics:\n",
       "%%!  %%HTML  %%SVG  %%bash  %%capture  %%debug  %%file  %%html  %%javascript  %%js  %%latex  %%markdown  %%perl  %%prun  %%pypy  %%python  %%python2  %%python3  %%ruby  %%script  %%sh  %%svg  %%sx  %%system  %%time  %%timeit  %%writefile\n",
       "\n",
       "Automagic is ON, % prefix IS NOT needed for line magics."
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "% lsmagic"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "mkdir /local/shared/pinsky_lab/sequencing/hiseq_2018_09_24_SEQ18\n",
    "mkdir /local/shared/pinsky_lab/sequencing/hiseq_2018_09_24_SEQ19\n",
    "mkdir /local/shared/pinsky_lab/sequencing/hiseq_2018_09_24_SEQ20\n",
    "mkdir /local/shared/pinsky_lab/sequencing/hiseq_2018_09_24_SEQ21"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Go to the Princeton webpage to retrieve the seq data - took about 5 minutes for SEQ18 - 576 samples\n",
    "https://htseq.princeton.edu/cgi-bin/login.pl?redirect_url=https://htseq.princeton.edu/cgi-bin/dashboard.pl **done**\n",
    "* Click on the sequencing run of interest in the box on the left that says “Recently Entered Samples\"\n",
    "* In the box titled Sample Provenance, click on the link following \"This library was utilized within the following output(s):” - repeat for each lane\n",
    "* In the “Data and Statistics” box, in the bottom right corner is a green button that says “Download\"\n",
    "* click wget and then the copy button to get all files.  This will copy the wget commands for all files.\n",
    "* paste in amphiprion **in the directory you made in the previous step**"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Make directories in personal workspace - 12 pools in each seq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd ~/02-apcl-ddocent\n",
    "mkdir 18seq 19seq 20seq 21seq\n",
    "cd ../18seq/ \n",
    "mkdir bcsplit 01Pool 02Pool 03Pool 04Pool 05Pool 06Pool 07Pool 08Pool 09Pool 10Pool 11Pool 12Pool logs samples scripts\n",
    "cd bcsplit/\n",
    "mkdir lane1 lane2\n",
    "cd ../../19seq/\n",
    "mkdir bcsplit 01Pool 02Pool 03Pool 04Pool 05Pool 06Pool 07Pool 08Pool 09Pool 10Pool 11Pool 12Pool logs samples scripts\n",
    "cd bcsplit/\n",
    "mkdir lane1 lane2\n",
    "cd ../../20seq/\n",
    "mkdir bcsplit 01Pool 02Pool 03Pool 04Pool 05Pool 06Pool 07Pool 08Pool 09Pool 10Pool 11Pool 12Pool logs samples scripts\n",
    "cd bcsplit/\n",
    "mkdir lane1 lane2\n",
    "cd ../../21seq/\n",
    "mkdir bcsplit 01Pool 02Pool 03Pool 04Pool 05Pool 06Pool 07Pool 08Pool 09Pool 10Pool 11Pool 12Pool logs samples scripts\n",
    "cd bcsplit/\n",
    "mkdir lane1 lane2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Move index files and names files from laptop to amphiprion with Fetch \n",
    "### see seq_proc.Rmd notebook in RStudio to generate index and names files for large batches of pools/samples"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Copy barcodes files into each of the logs folders"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd ~/02-apcl-ddocent\n",
    "cp 16seq/logs/barcodes 18seq/logs/ \n",
    "cp 16seq/logs/barcodes 19seq/logs \n",
    "cp 16seq/logs/barcodes 20seq/logs \n",
    "cp 16seq/logs/barcodes 21seq/logs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# SEQ18_APCL_MRS"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Run barcode splitter in lane1 folder and lane 2 folder - takes about 8 hours for 192 samples, \n",
    "started 10am on Monday, finished at about 7:30pm on Monday - took about 10 hours for 576 samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd ~/02-apcl-ddocent/18seq/bcsplit/lane1\n",
    "nohup barcode_splitter.py --bcfile ../../logs/index-seq18.tsv --idxread 2 --suffix .fastq.gz /local/shared/pinsky_lab/sequencing/hiseq_2018_09_24_SEQ18/18_APCL_SEQ-for-196-cycles-HLMHLBCX2_1_Read_1_passed_filter.fastq.gz /local/shared/pinsky_lab/sequencing/hiseq_2018_09_24_SEQ18/18_APCL_SEQ-for-196-cycles-HLMHLBCX2_1_Read_2_Index_Read_passed_filter.fastq.gz\n",
    "cd ~/02-apcl-ddocent/18seq/bcsplit/lane2\n",
    "nohup barcode_splitter.py --bcfile ../../logs/index-seq18.tsv --idxread 2 --suffix .fastq.gz /local/shared/pinsky_lab/sequencing/hiseq_2018_09_24_SEQ18/18_APCL_SEQ-for-196-cycles-HLMHLBCX2_2_Read_1_passed_filter.fastq.gz /local/shared/pinsky_lab/sequencing/hiseq_2018_09_24_SEQ18/18_APCL_SEQ-for-196-cycles-HLMHLBCX2_2_Read_2_Index_Read_passed_filter.fastq.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Took about 10 hours and there was no output text in nohup.out and no stats."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Concatenate the results in the bcsplit folder - takes about 1 minute\n",
    "For large batches, generated the command lines in a script in the seq_proc.Rmd notebook and output the lines into files called SEQXX_cat_all.sh to be copy and pasted here\n",
    "# For SEQ18 the pool order kept getting rearranged to be P075, P076, P073, P074 and so these samples might have gotten rearranged.  I manuallly changed the output before pasting the lines as you see them below.  I think that everything is correct because the correct index code was assigned to the correct pool and so anything with that index would have been put in the correct pool folder, but if there are future problems, that is the most likely source"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cat ./lane1/P075-read-1.fastq.gz ./lane2/P075-read-1.fastq.gz > ../03Pool/P075.fastq.gz\n",
    "cat ./lane1/P076-read-1.fastq.gz ./lane2/P076-read-1.fastq.gz > ../04Pool/P076.fastq.gz\n",
    "cat ./lane1/P073-read-1.fastq.gz ./lane2/P073-read-1.fastq.gz > ../01Pool/P073.fastq.gz\n",
    "cat ./lane1/P074-read-1.fastq.gz ./lane2/P074-read-1.fastq.gz > ../02Pool/P074.fastq.gz\n",
    "cat ./lane1/P077-read-1.fastq.gz ./lane2/P077-read-1.fastq.gz > ../05Pool/P077.fastq.gz\n",
    "cat ./lane1/P078-read-1.fastq.gz ./lane2/P078-read-1.fastq.gz > ../06Pool/P078.fastq.gz\n",
    "cat ./lane1/P083-read-1.fastq.gz ./lane2/P083-read-1.fastq.gz > ../07Pool/P083.fastq.gz\n",
    "cat ./lane1/P084-read-1.fastq.gz ./lane2/P084-read-1.fastq.gz > ../08Pool/P084.fastq.gz\n",
    "cat ./lane1/P085-read-1.fastq.gz ./lane2/P085-read-1.fastq.gz > ../09Pool/P085.fastq.gz\n",
    "cat ./lane1/P086-read-1.fastq.gz ./lane2/P086-read-1.fastq.gz > ../10Pool/P086.fastq.gz\n",
    "cat ./lane1/P087-read-1.fastq.gz ./lane2/P087-read-1.fastq.gz > ../11Pool/P087.fastq.gz\n",
    "cat ./lane1/P088-read-1.fastq.gz ./lane2/P088-read-1.fastq.gz > ../12Pool/P088.fastq.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Created process_radtags scripts with the seq_proc.Rmd notebook and then moved the files with fetch to amphiprion in the 18seq/scripts folder.  In fetch, also highlighted the scripts, clicked get info, and then checked \"execute\" instead of having to chmod u+x for all of them.\n",
    "\n",
    "## Also wrote a script to generate the command lines to run process rad tags.  These need to be run in separate windows (not sure how that works through a jupyter notebook).  The command lines are in the files SEQXX_process_all.sh\n",
    "\n",
    "## Started at 9:45pm on Monday night for SEQ18, finished before 6am on Tuesday."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "nohup ./scripts/P073_process.sh &\n",
    "nohup ./scripts/P074_process.sh &\n",
    "nohup ./scripts/P075_process.sh &\n",
    "nohup ./scripts/P076_process.sh &\n",
    "nohup ./scripts/P077_process.sh &\n",
    "nohup ./scripts/P078_process.sh &\n",
    "nohup ./scripts/P083_process.sh &\n",
    "nohup ./scripts/P084_process.sh &\n",
    "nohup ./scripts/P085_process.sh &\n",
    "nohup ./scripts/P086_process.sh &\n",
    "nohup ./scripts/P087_process.sh &\n",
    "nohup ./scripts/P088_process.sh &"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Move the process_radtags logs to the logs folder - wrote command lines for large batches in seq_proc.Rmd notebook and saved the lines in files called SEQXX-move_radlogs.sh "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "mv 01Pool/process_radtags.log ./logs/processP075.log\n",
    "mv 02Pool/process_radtags.log ./logs/processP076.log\n",
    "mv 03Pool/process_radtags.log ./logs/processP073.log\n",
    "mv 04Pool/process_radtags.log ./logs/processP074.log\n",
    "mv 05Pool/process_radtags.log ./logs/processP077.log\n",
    "mv 06Pool/process_radtags.log ./logs/processP078.log\n",
    "mv 07Pool/process_radtags.log ./logs/processP083.log\n",
    "mv 08Pool/process_radtags.log ./logs/processP084.log\n",
    "mv 09Pool/process_radtags.log ./logs/processP085.log\n",
    "mv 10Pool/process_radtags.log ./logs/processP086.log\n",
    "mv 11Pool/process_radtags.log ./logs/processP087.log\n",
    "mv 12Pool/process_radtags.log ./logs/processP088.log\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# it looks like I named the P073-P076 logs to the wrong names - fixing it here"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/local/home/michelles/02-apcl-ddocent/18seq/logs\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd ~/02-apcl-ddocent/18seq/logs/\n",
    "pwd\n",
    "mv processP075.log processP073.log.b\n",
    "mv processP073.log processP075.log\n",
    "mv processP073.log.b processP073.log\n",
    "mv processP076.log processP074.log.b\n",
    "mv processP074.log processP076.log\n",
    "mv processP074.log.b processP074.log"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Run the readprocesslog.py script to convert the process log into a tsv that can be imported into the database through R"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AAACAC\t184410\t544\t2287\t180651\n",
      "AAACGA\t3049\t12\t64\t2959\n",
      "AAAGTC\t254\t3\t10\t240\n",
      "AACGGT\t424\t4\t35\t383\n",
      "AACTTC\t42899\t135\t595\t41979\n",
      "TGCTCA\t294126\t777\t3494\t288375\n",
      "AAGAAC\t15143\t48\t196\t14828\n",
      "AATGTG\t348810\t1739\t4301\t340906\n",
      "ACATGT\t128123\t529\t1634\t125283\n",
      "ACCAAA\t135301\t390\t1787\t132476\n",
      "ACGATA\t307763\t781\t3946\t301508\n",
      "ACGTTT\t411092\t4607\t5438\t399045\n",
      "ACTAGG\t107275\t431\t1376\t104846\n",
      "ACTCCA\t135777\t384\t1662\t133027\n",
      "AGACTC\t554874\t1445\t6934\t543964\n",
      "AGCATT\t73469\t356\t961\t71750\n",
      "AGGAGA\t180997\t583\t2261\t177258\n",
      "AGTAAG\t295916\t1038\t3664\t289711\n",
      "AGTCCT\t106254\t398\t1387\t103968\n",
      "AGTTAC\t284683\t915\t3885\t278372\n",
      "ATAACC\t495362\t2550\t6621\t483717\n",
      "ATCGCA\t505607\t1482\t6762\t494689\n",
      "ATCTCG\t609247\t1763\t8079\t596235\n",
      "ATGGAG\t180117\t649\t2292\t176142\n",
      "ATGTCC\t381\t5\t31\t343\n",
      "CAATTG\t375559\t1880\t4581\t367250\n",
      "CAGAGT\t1216\t15\t33\t1164\n",
      "CATCAG\t577\t16\t15\t543\n",
      "CATCTC\t339022\t1180\t4152\t331830\n",
      "CCACTT\t622318\t3002\t8066\t608307\n",
      "CCCATA\t611199\t1664\t7833\t598970\n",
      "CCTGAA\t444268\t1317\t5615\t435273\n",
      "CGAAAC\t150051\t431\t1810\t147086\n",
      "CGAATG\t4101\t87\t61\t3931\n",
      "GACGTT\t279543\t977\t3571\t273749\n",
      "GATACA\t47201\t120\t638\t46232\n",
      "GCAGAA\t352914\t1012\t4271\t345785\n",
      "GGGATA\t151676\t558\t1844\t148573\n",
      "GGTGAA\t141935\t495\t1600\t139112\n",
      "GTAGCT\t421092\t1292\t5228\t412733\n",
      "GTCTAT\t126800\t381\t1576\t124240\n",
      "GTTCAG\t263050\t911\t2994\t257900\n",
      "TAAGAC\t164730\t486\t1923\t161403\n",
      "TACCAG\t286207\t1014\t3555\t280324\n",
      "TCAATC\t345607\t1161\t4121\t338703\n",
      "TCCAAA\t269829\t732\t3331\t264393\n",
      "TCTGCT\t257406\t741\t3099\t252326\n",
      "TCTTAG\t51885\t1289\t641\t49741\n"
     ]
    }
   ],
   "source": [
    "# latlon_3.py - for use in Chapter 10 PCfB\n",
    "# Read in each line of the example file, split it into \n",
    "# separate components, and write certain output to a separate file\n",
    "\n",
    "# Set the input file name\n",
    "# (The program must be run from within the directory \n",
    "#  that contains this data file)\n",
    "#InFileName = './logs/16process.out'\n",
    "InFileName = '/local/home/michelles/02-apcl-ddocent/18seq/logs/processP073.log'\n",
    "\n",
    "# Open the input file for reading\n",
    "InFile = open(InFileName, 'r')\n",
    "\n",
    "# Initialize the counter used to keep track of line numbers\n",
    "LineNumber = 0\n",
    "\n",
    "# Open the output file for writing\n",
    "# Do this *before* the loop, not inside it\n",
    "OutFileName=InFileName + \".tsv\"\n",
    "\n",
    "OutFile=open(OutFileName,'w') # You can append instead with 'a'\n",
    "\n",
    "# Loop through each line in the file\n",
    "for Line in InFile:\n",
    "        # Skip the header, line # 0\n",
    "        if LineNumber < 62:\n",
    "                if LineNumber > 13:\n",
    "                        # Remove the line ending characters\n",
    "                        Line=Line.strip('\\n')\n",
    "                        \n",
    "                        # Separate the line into a list of its tab-delimited components\n",
    "                        ElementList=Line.split('\\t')\n",
    "                        Barcode   = ElementList[0]\n",
    "                        Total_Reads = ElementList[1]\n",
    "                        Lack_RadTag = ElementList[2]  \n",
    "                        Low_Quality = ElementList[3]\n",
    "                        Retained = ElementList[4]\n",
    "                        \n",
    "                        # Use the % operator to generate a string \n",
    "                        # We can use this for output both to the screen and to a file\n",
    "                        OutputString = \"%s\\t%s\\t%s\\t%s\\t%s\" % (ElementList[0], ElementList[1], ElementList[2], ElementList[3], ElementList[4])\n",
    "                           \n",
    "                        # Can still print to the screen then write to a file\n",
    "                        print (OutputString)\n",
    "                        \n",
    "                        # Unlike print statements, .write needs a linefeed\n",
    "                        OutFile.write(OutputString+\"\\n\")  \n",
    "                \n",
    "        # Index the counter used to keep track of line numbers\n",
    "        LineNumber = LineNumber + 1\n",
    "        # After the loop is completed, close the files\n",
    "InFile.close()\n",
    "OutFile.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "barcodes         names_078.tsv  processP073.log      processP083.log\r\n",
      "index-seq18.tsv  names_083.tsv  processP073.log.tsv  processP084.log\r\n",
      "names_073.tsv    names_084.tsv  processP074.log      processP085.log\r\n",
      "names_074.tsv    names_085.tsv  processP075.log      processP086.log\r\n",
      "names_075.tsv    names_086.tsv  processP076.log      processP087.log\r\n",
      "names_076.tsv    names_087.tsv  processP077.log      processP088.log\r\n",
      "names_077.tsv    names_088.tsv  processP078.log\r\n"
     ]
    }
   ],
   "source": [
    "%ls 02-apcl-ddocent/18seq/logs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AAACAC\t286581\t921\t3986\t280149\n",
      "AAACGA\t23464\t68\t319\t22969\n",
      "AAAGTC\t308347\t1000\t3847\t301991\n",
      "AACGGT\t479304\t2311\t6190\t468136\n",
      "AACTTC\t402800\t1285\t5759\t393856\n",
      "TGCTCA\t297985\t807\t3764\t291881\n",
      "AAGAAC\t279586\t1145\t3489\t273624\n",
      "AATGTG\t143766\t590\t1948\t140574\n",
      "ACATGT\t366207\t1493\t4770\t357945\n",
      "ACCAAA\t458046\t1765\t6234\t447605\n",
      "ACGATA\t48343\t218\t629\t47232\n",
      "ACGTTT\t405859\t2780\t5756\t395044\n",
      "ACTAGG\t257218\t907\t3284\t251646\n",
      "ACTCCA\t323123\t1445\t4256\t315915\n",
      "AGACTC\t165108\t689\t2063\t161512\n",
      "AGCATT\t606406\t2667\t7970\t592646\n",
      "AGGAGA\t429575\t1223\t5382\t420463\n",
      "AGTAAG\t140100\t436\t1752\t137101\n",
      "AGTCCT\t513189\t2545\t6601\t501268\n",
      "AGTTAC\t350908\t1302\t4749\t342896\n",
      "ATAACC\t278692\t950\t3867\t272383\n",
      "ATCGCA\t127602\t323\t1774\t124838\n",
      "ATCTCG\t508599\t1338\t6918\t498000\n",
      "ATGGAG\t275501\t1085\t3708\t269080\n",
      "ATGTCC\t345918\t941\t4513\t338782\n",
      "CAATTG\t211679\t899\t2768\t206858\n",
      "CAGAGT\t441716\t2067\t5447\t431750\n",
      "CATCAG\t25657\t154\t369\t24959\n",
      "CATCTC\t4876\t23\t78\t4750\n",
      "CCACTT\t392079\t1505\t5332\t383154\n",
      "CCCATA\t610451\t1696\t7990\t598219\n",
      "CCTGAA\t488638\t1662\t5860\t478629\n",
      "CGAAAC\t188222\t625\t2296\t184383\n",
      "CGAATG\t190016\t1082\t2401\t185445\n",
      "GACGTT\t136772\t645\t1755\t133741\n",
      "GATACA\t118510\t382\t1563\t115855\n",
      "GCAGAA\t591034\t2100\t7099\t578419\n",
      "GGGATA\t460005\t1615\t6137\t449902\n",
      "GGTGAA\t575240\t2448\t7007\t562910\n",
      "GTAGCT\t457521\t1600\t6085\t447439\n",
      "GTCTAT\t318993\t1050\t3871\t312431\n",
      "GTTCAG\t121644\t529\t1439\t118990\n",
      "TAAGAC\t595626\t1908\t7236\t583349\n",
      "TACCAG\t247707\t1443\t3411\t241487\n",
      "TCAATC\t34009\t120\t438\t33274\n",
      "TCCAAA\t235048\t626\t3032\t230274\n",
      "TCTGCT\t439300\t1475\t5491\t429869\n",
      "TCTTAG\t256021\t1537\t3131\t250179\n"
     ]
    }
   ],
   "source": [
    "# latlon_3.py - for use in Chapter 10 PCfB\n",
    "# Read in each line of the example file, split it into \n",
    "# separate components, and write certain output to a separate file\n",
    "\n",
    "# Set the input file name\n",
    "# (The program must be run from within the directory \n",
    "#  that contains this data file)\n",
    "#InFileName = './logs/16process.out'\n",
    "InFileName = '/local/home/michelles/02-apcl-ddocent/18seq/logs/processP074.log'\n",
    "\n",
    "# Open the input file for reading\n",
    "InFile = open(InFileName, 'r')\n",
    "\n",
    "# Initialize the counter used to keep track of line numbers\n",
    "LineNumber = 0\n",
    "\n",
    "# Open the output file for writing\n",
    "# Do this *before* the loop, not inside it\n",
    "OutFileName=InFileName + \".tsv\"\n",
    "\n",
    "OutFile=open(OutFileName,'w') # You can append instead with 'a'\n",
    "\n",
    "# Loop through each line in the file\n",
    "for Line in InFile:\n",
    "        # Skip the header, line # 0\n",
    "        if LineNumber < 62:\n",
    "                if LineNumber > 13:\n",
    "                        # Remove the line ending characters\n",
    "                        Line=Line.strip('\\n')\n",
    "                        \n",
    "                        # Separate the line into a list of its tab-delimited components\n",
    "                        ElementList=Line.split('\\t')\n",
    "                        Barcode   = ElementList[0]\n",
    "                        Total_Reads = ElementList[1]\n",
    "                        Lack_RadTag = ElementList[2]  \n",
    "                        Low_Quality = ElementList[3]\n",
    "                        Retained = ElementList[4]\n",
    "                        \n",
    "                        # Use the % operator to generate a string \n",
    "                        # We can use this for output both to the screen and to a file\n",
    "                        OutputString = \"%s\\t%s\\t%s\\t%s\\t%s\" % (ElementList[0], ElementList[1], ElementList[2], ElementList[3], ElementList[4])\n",
    "                           \n",
    "                        # Can still print to the screen then write to a file\n",
    "                        print (OutputString)\n",
    "                        \n",
    "                        # Unlike print statements, .write needs a linefeed\n",
    "                        OutFile.write(OutputString+\"\\n\")  \n",
    "                \n",
    "        # Index the counter used to keep track of line numbers\n",
    "        LineNumber = LineNumber + 1\n",
    "        # After the loop is completed, close the files\n",
    "InFile.close()\n",
    "OutFile.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AAACAC\t611438\t4141\t10121\t593862\n",
      "AAACGA\t685006\t3842\t11635\t665387\n",
      "AAAGTC\t651189\t2848\t9889\t634697\n",
      "AACGGT\t818631\t3567\t13299\t796202\n",
      "AACTTC\t715900\t2762\t12630\t696563\n",
      "TGCTCA\t497017\t2607\t7865\t483722\n",
      "AAGAAC\t672194\t2900\t9886\t655363\n",
      "AATGTG\t805988\t3400\t12012\t785724\n",
      "ACATGT\t280651\t1098\t4081\t273974\n",
      "ACCAAA\t391728\t1370\t5957\t381880\n",
      "ACGATA\t359954\t1441\t5385\t351155\n",
      "ACGTTT\t468370\t3467\t7615\t455034\n",
      "ACTAGG\t251822\t1672\t3686\t245072\n",
      "ACTCCA\t271383\t950\t3958\t265106\n",
      "AGACTC\t286043\t817\t4167\t279549\n",
      "AGCATT\t766704\t3560\t11513\t747352\n",
      "AGGAGA\t627866\t2076\t8778\t613139\n",
      "AGTAAG\t527695\t1991\t7728\t514689\n",
      "AGTCCT\t318912\t2799\t4576\t309999\n",
      "AGTTAC\t337237\t911\t5149\t329195\n",
      "ATAACC\t378399\t1238\t5963\t369043\n",
      "ATCGCA\t566520\t1612\t8858\t552920\n",
      "ATCTCG\t327208\t949\t5029\t319498\n",
      "ATGGAG\t666548\t2232\t10130\t649593\n",
      "ATGTCC\t827845\t2202\t12229\t808739\n",
      "CAATTG\t705520\t2614\t10147\t688504\n",
      "CAGAGT\t832933\t4021\t12680\t811129\n",
      "CATCAG\t580259\t2288\t8004\t566658\n",
      "CATCTC\t454706\t1210\t6512\t443903\n",
      "CCACTT\t962338\t3176\t13927\t939838\n",
      "CCCATA\t622992\t1809\t9331\t608492\n",
      "CCTGAA\t548341\t1798\t8233\t535321\n",
      "CGAAAC\t835921\t2532\t11880\t816704\n",
      "CGAATG\t847102\t4283\t14240\t823436\n",
      "GACGTT\t904852\t3435\t13166\t883126\n",
      "GATACA\t728563\t1949\t10416\t711878\n",
      "GCAGAA\t824197\t2419\t10663\t806151\n",
      "GGGATA\t729056\t2369\t11009\t711963\n",
      "GGTGAA\t444854\t1543\t6113\t434672\n",
      "GTAGCT\t483419\t1901\t7447\t471315\n",
      "GTCTAT\t341803\t1427\t4861\t333750\n",
      "GTTCAG\t423\t5\t20\t396\n",
      "TAAGAC\t593\t6\t46\t540\n",
      "TACCAG\t483320\t1526\t7484\t471652\n",
      "TCAATC\t564\t13\t34\t514\n",
      "TCCAAA\t607422\t1589\t8574\t593552\n",
      "TCTGCT\t508928\t1395\t7333\t497302\n",
      "TCTTAG\t751368\t2129\t10222\t734644\n"
     ]
    }
   ],
   "source": [
    "# latlon_3.py - for use in Chapter 10 PCfB\n",
    "# Read in each line of the example file, split it into \n",
    "# separate components, and write certain output to a separate file\n",
    "\n",
    "# Set the input file name\n",
    "# (The program must be run from within the directory \n",
    "#  that contains this data file)\n",
    "#InFileName = './logs/16process.out'\n",
    "InFileName = '/local/home/michelles/02-apcl-ddocent/18seq/logs/processP075.log'\n",
    "\n",
    "# Open the input file for reading\n",
    "InFile = open(InFileName, 'r')\n",
    "\n",
    "# Initialize the counter used to keep track of line numbers\n",
    "LineNumber = 0\n",
    "\n",
    "# Open the output file for writing\n",
    "# Do this *before* the loop, not inside it\n",
    "OutFileName=InFileName + \".tsv\"\n",
    "\n",
    "OutFile=open(OutFileName,'w') # You can append instead with 'a'\n",
    "\n",
    "# Loop through each line in the file\n",
    "for Line in InFile:\n",
    "        # Skip the header, line # 0\n",
    "        if LineNumber < 62:\n",
    "                if LineNumber > 13:\n",
    "                        # Remove the line ending characters\n",
    "                        Line=Line.strip('\\n')\n",
    "                        \n",
    "                        # Separate the line into a list of its tab-delimited components\n",
    "                        ElementList=Line.split('\\t')\n",
    "                        Barcode   = ElementList[0]\n",
    "                        Total_Reads = ElementList[1]\n",
    "                        Lack_RadTag = ElementList[2]  \n",
    "                        Low_Quality = ElementList[3]\n",
    "                        Retained = ElementList[4]\n",
    "                        \n",
    "                        # Use the % operator to generate a string \n",
    "                        # We can use this for output both to the screen and to a file\n",
    "                        OutputString = \"%s\\t%s\\t%s\\t%s\\t%s\" % (ElementList[0], ElementList[1], ElementList[2], ElementList[3], ElementList[4])\n",
    "                           \n",
    "                        # Can still print to the screen then write to a file\n",
    "                        print (OutputString)\n",
    "                        \n",
    "                        # Unlike print statements, .write needs a linefeed\n",
    "                        OutFile.write(OutputString+\"\\n\")  \n",
    "                \n",
    "        # Index the counter used to keep track of line numbers\n",
    "        LineNumber = LineNumber + 1\n",
    "        # After the loop is completed, close the files\n",
    "InFile.close()\n",
    "OutFile.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AAACAC\t1047823\t5454\t13943\t1022118\n",
      "AAACGA\t1238925\t3664\t15965\t1212020\n",
      "AAAGTC\t771624\t2346\t9580\t755320\n",
      "AACGGT\t1002155\t3668\t13324\t979004\n",
      "AACTTC\t2749\t60\t213\t2471\n",
      "TGCTCA\t2821\t25\t239\t2550\n",
      "AAGAAC\t2774\t18\t222\t2524\n",
      "AATGTG\t35202\t187\t2790\t32075\n",
      "ACATGT\t973979\t3921\t12593\t952103\n",
      "ACCAAA\t1012023\t2716\t12949\t990138\n",
      "ACGATA\t35113\t116\t2775\t32096\n",
      "ACGTTT\t23866\t1486\t1918\t20382\n",
      "ACTAGG\t21404\t91\t1613\t19614\n",
      "ACTCCA\t771\t9\t35\t718\n",
      "AGACTC\t535\t7\t46\t477\n",
      "AGCATT\t554\t11\t36\t504\n",
      "AGGAGA\t1133769\t5164\t15330\t1105932\n",
      "AGTAAG\t845226\t4554\t11366\t824448\n",
      "AGTCCT\t793167\t4206\t10567\t773785\n",
      "AGTTAC\t968935\t3776\t12999\t946571\n",
      "ATAACC\t622834\t3332\t9104\t606666\n",
      "ATCGCA\t855875\t2666\t11979\t836175\n",
      "ATCTCG\t930013\t2744\t12374\t909872\n",
      "ATGGAG\t605620\t5025\t8756\t587930\n",
      "ATGTCC\t521318\t1301\t6648\t510550\n",
      "CAATTG\t569576\t2059\t6986\t556659\n",
      "CAGAGT\t953165\t4922\t11078\t931424\n",
      "CATCAG\t797494\t2833\t9372\t780013\n",
      "CATCTC\t546\t12\t36\t495\n",
      "CCACTT\t749\t17\t50\t676\n",
      "CCCATA\t577529\t1663\t7473\t565652\n",
      "CCTGAA\t701282\t2224\t7966\t686899\n",
      "CGAAAC\t579\t21\t33\t524\n",
      "CGAATG\t819\t124\t40\t653\n",
      "GACGTT\t1067184\t4025\t13204\t1043482\n",
      "GATACA\t406660\t1795\t5237\t397203\n",
      "GCAGAA\t700544\t2210\t8086\t686036\n",
      "GGGATA\t1048007\t3598\t13657\t1024735\n",
      "GGTGAA\t618552\t2309\t7593\t605089\n",
      "GTAGCT\t432274\t1579\t5573\t422363\n",
      "GTCTAT\t580355\t2285\t7355\t567342\n",
      "GTTCAG\t448188\t1620\t5633\t438374\n",
      "TAAGAC\t573131\t2356\t7185\t560229\n",
      "TACCAG\t689061\t2110\t8968\t674511\n",
      "TCAATC\t836841\t2368\t10194\t819587\n",
      "TCCAAA\t720641\t2198\t9356\t704772\n",
      "TCTGCT\t1051618\t3394\t13143\t1028691\n",
      "TCTTAG\t611\t9\t33\t565\n"
     ]
    }
   ],
   "source": [
    "# latlon_3.py - for use in Chapter 10 PCfB\n",
    "# Read in each line of the example file, split it into \n",
    "# separate components, and write certain output to a separate file\n",
    "\n",
    "# Set the input file name\n",
    "# (The program must be run from within the directory \n",
    "#  that contains this data file)\n",
    "#InFileName = './logs/16process.out'\n",
    "InFileName = '/local/home/michelles/02-apcl-ddocent/18seq/logs/processP076.log'\n",
    "\n",
    "# Open the input file for reading\n",
    "InFile = open(InFileName, 'r')\n",
    "\n",
    "# Initialize the counter used to keep track of line numbers\n",
    "LineNumber = 0\n",
    "\n",
    "# Open the output file for writing\n",
    "# Do this *before* the loop, not inside it\n",
    "OutFileName=InFileName + \".tsv\"\n",
    "\n",
    "OutFile=open(OutFileName,'w') # You can append instead with 'a'\n",
    "\n",
    "# Loop through each line in the file\n",
    "for Line in InFile:\n",
    "        # Skip the header, line # 0\n",
    "        if LineNumber < 62:\n",
    "                if LineNumber > 13:\n",
    "                        # Remove the line ending characters\n",
    "                        Line=Line.strip('\\n')\n",
    "                        \n",
    "                        # Separate the line into a list of its tab-delimited components\n",
    "                        ElementList=Line.split('\\t')\n",
    "                        Barcode   = ElementList[0]\n",
    "                        Total_Reads = ElementList[1]\n",
    "                        Lack_RadTag = ElementList[2]  \n",
    "                        Low_Quality = ElementList[3]\n",
    "                        Retained = ElementList[4]\n",
    "                        \n",
    "                        # Use the % operator to generate a string \n",
    "                        # We can use this for output both to the screen and to a file\n",
    "                        OutputString = \"%s\\t%s\\t%s\\t%s\\t%s\" % (ElementList[0], ElementList[1], ElementList[2], ElementList[3], ElementList[4])\n",
    "                           \n",
    "                        # Can still print to the screen then write to a file\n",
    "                        print (OutputString)\n",
    "                        \n",
    "                        # Unlike print statements, .write needs a linefeed\n",
    "                        OutFile.write(OutputString+\"\\n\")  \n",
    "                \n",
    "        # Index the counter used to keep track of line numbers\n",
    "        LineNumber = LineNumber + 1\n",
    "        # After the loop is completed, close the files\n",
    "InFile.close()\n",
    "OutFile.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AAACAC\t636159\t4899\t8699\t618617\n",
      "AAACGA\t427099\t2267\t6114\t416005\n",
      "AAAGTC\t303089\t1467\t3930\t295944\n",
      "AACGGT\t1110827\t4831\t14924\t1084381\n",
      "AACTTC\t516713\t1693\t7031\t505042\n",
      "TGCTCA\t1004575\t4143\t13129\t981022\n",
      "AAGAAC\t579584\t2534\t7417\t565913\n",
      "AATGTG\t1191666\t5419\t15706\t1163147\n",
      "ACATGT\t831459\t3139\t10900\t812851\n",
      "ACCAAA\t421573\t2413\t5826\t410872\n",
      "ACGATA\t640909\t4742\t9333\t623418\n",
      "ACGTTT\t723047\t6105\t11413\t701605\n",
      "ACTAGG\t1008\t11\t95\t901\n",
      "ACTCCA\t1948\t41\t39\t1861\n",
      "AGACTC\t411425\t1890\t5784\t401386\n",
      "AGCATT\t350910\t1755\t4615\t342020\n",
      "AGGAGA\t317664\t2382\t4661\t308949\n",
      "AGTAAG\t398126\t2624\t5690\t387445\n",
      "AGTCCT\t312060\t3147\t4521\t302668\n",
      "AGTTAC\t397331\t2242\t6038\t386808\n",
      "ATAACC\t361194\t1979\t5607\t351608\n",
      "ATCGCA\t625891\t2659\t9017\t610975\n",
      "ATCTCG\t659483\t2112\t8704\t644936\n",
      "ATGGAG\t382261\t1577\t5060\t373059\n",
      "ATGTCC\t627891\t2104\t8108\t613857\n",
      "CAATTG\t507492\t2935\t6641\t494954\n",
      "CAGAGT\t343412\t2215\t4328\t334483\n",
      "CATCAG\t448209\t1587\t5366\t438761\n",
      "CATCTC\t422206\t1317\t5355\t413202\n",
      "CCACTT\t444\t24\t35\t385\n",
      "CCCATA\t555\t15\t38\t501\n",
      "CCTGAA\t542810\t1805\t7002\t530831\n",
      "CGAAAC\t365\t19\t23\t319\n",
      "CGAATG\t222646\t969\t2980\t217303\n",
      "GACGTT\t489227\t2088\t6396\t478007\n",
      "GATACA\t386868\t1170\t5141\t378185\n",
      "GCAGAA\t681266\t3730\t8891\t664392\n",
      "GGGATA\t10900\t75\t872\t9914\n",
      "GGTGAA\t15070\t93\t1102\t13826\n",
      "GTAGCT\t21087\t97\t1737\t19196\n",
      "GTCTAT\t15769\t94\t1189\t14437\n",
      "GTTCAG\t9266\t67\t668\t8514\n",
      "TAAGAC\t17907\t81\t1406\t16372\n",
      "TACCAG\t29745\t170\t2232\t27226\n",
      "TCAATC\t24400\t100\t1820\t22376\n",
      "TCCAAA\t33577\t154\t2543\t30765\n",
      "TCTGCT\t26630\t138\t2107\t24299\n",
      "TCTTAG\t46094\t279\t3519\t42157\n"
     ]
    }
   ],
   "source": [
    "# latlon_3.py - for use in Chapter 10 PCfB\n",
    "# Read in each line of the example file, split it into \n",
    "# separate components, and write certain output to a separate file\n",
    "\n",
    "# Set the input file name\n",
    "# (The program must be run from within the directory \n",
    "#  that contains this data file)\n",
    "#InFileName = './logs/16process.out'\n",
    "InFileName = '/local/home/michelles/02-apcl-ddocent/18seq/logs/processP077.log'\n",
    "\n",
    "# Open the input file for reading\n",
    "InFile = open(InFileName, 'r')\n",
    "\n",
    "# Initialize the counter used to keep track of line numbers\n",
    "LineNumber = 0\n",
    "\n",
    "# Open the output file for writing\n",
    "# Do this *before* the loop, not inside it\n",
    "OutFileName=InFileName + \".tsv\"\n",
    "\n",
    "OutFile=open(OutFileName,'w') # You can append instead with 'a'\n",
    "\n",
    "# Loop through each line in the file\n",
    "for Line in InFile:\n",
    "        # Skip the header, line # 0\n",
    "        if LineNumber < 62:\n",
    "                if LineNumber > 13:\n",
    "                        # Remove the line ending characters\n",
    "                        Line=Line.strip('\\n')\n",
    "                        \n",
    "                        # Separate the line into a list of its tab-delimited components\n",
    "                        ElementList=Line.split('\\t')\n",
    "                        Barcode   = ElementList[0]\n",
    "                        Total_Reads = ElementList[1]\n",
    "                        Lack_RadTag = ElementList[2]  \n",
    "                        Low_Quality = ElementList[3]\n",
    "                        Retained = ElementList[4]\n",
    "                        \n",
    "                        # Use the % operator to generate a string \n",
    "                        # We can use this for output both to the screen and to a file\n",
    "                        OutputString = \"%s\\t%s\\t%s\\t%s\\t%s\" % (ElementList[0], ElementList[1], ElementList[2], ElementList[3], ElementList[4])\n",
    "                           \n",
    "                        # Can still print to the screen then write to a file\n",
    "                        print (OutputString)\n",
    "                        \n",
    "                        # Unlike print statements, .write needs a linefeed\n",
    "                        OutFile.write(OutputString+\"\\n\")  \n",
    "                \n",
    "        # Index the counter used to keep track of line numbers\n",
    "        LineNumber = LineNumber + 1\n",
    "        # After the loop is completed, close the files\n",
    "InFile.close()\n",
    "OutFile.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AAACAC\t308115\t1306\t4082\t300852\n",
      "AAACGA\t602781\t2159\t8593\t588508\n",
      "AAAGTC\t303177\t986\t3938\t296583\n",
      "AACGGT\t305261\t1253\t4196\t298102\n",
      "AACTTC\t140407\t363\t1958\t137227\n",
      "TGCTCA\t668158\t1694\t8516\t653911\n",
      "AAGAAC\t446683\t1334\t5895\t436974\n",
      "AATGTG\t475148\t1871\t6373\t464245\n",
      "ACATGT\t525840\t1777\t7042\t514161\n",
      "ACCAAA\t843217\t2241\t11719\t824078\n",
      "ACGATA\t488436\t1399\t6625\t477383\n",
      "ACGTTT\t494422\t2939\t7231\t481045\n",
      "ACTAGG\t464111\t1531\t6028\t453664\n",
      "ACTCCA\t761947\t2129\t10591\t745546\n",
      "AGACTC\t315290\t1197\t4366\t308132\n",
      "AGCATT\t232587\t1476\t3377\t226490\n",
      "AGGAGA\t378\t10\t31\t336\n",
      "AGTAAG\t286721\t1525\t4043\t279511\n",
      "AGTCCT\t341727\t2100\t4681\t333044\n",
      "AGTTAC\t411067\t1040\t5543\t402164\n",
      "ATAACC\t430747\t1929\t5976\t420578\n",
      "ATCGCA\t215092\t674\t2978\t210345\n",
      "ATCTCG\t426178\t983\t5763\t417256\n",
      "ATGGAG\t204020\t653\t2663\t199682\n",
      "ATGTCC\t248990\t741\t3393\t243597\n",
      "CAATTG\t320745\t1162\t3995\t313534\n",
      "CAGAGT\t480330\t2440\t6418\t468654\n",
      "CATCAG\t344732\t1100\t4363\t337345\n",
      "CATCTC\t307\t6\t18\t281\n",
      "CCACTT\t467132\t1486\t6285\t456754\n",
      "CCCATA\t336442\t907\t4628\t329158\n",
      "CCTGAA\t166321\t478\t2252\t162766\n",
      "CGAAAC\t306655\t835\t4168\t299820\n",
      "CGAATG\t424642\t1367\t5516\t415344\n",
      "GACGTT\t561172\t2031\t7285\t548587\n",
      "GATACA\t433361\t1365\t5561\t424188\n",
      "GCAGAA\t520869\t1530\t6279\t510169\n",
      "GGGATA\t11894\t78\t921\t10866\n",
      "GGTGAA\t19257\t70\t1528\t17591\n",
      "GTAGCT\t707\t8\t52\t644\n",
      "GTCTAT\t16262\t70\t1296\t14849\n",
      "GTTCAG\t19122\t69\t1431\t17567\n",
      "TAAGAC\t5852\t30\t422\t5380\n",
      "TACCAG\t271\t5\t10\t253\n",
      "TCAATC\t401\t12\t16\t368\n",
      "TCCAAA\t13442\t49\t1020\t12330\n",
      "TCTGCT\t163134\t480\t2095\t159592\n",
      "TCTTAG\t257357\t796\t3184\t251963\n"
     ]
    }
   ],
   "source": [
    "# latlon_3.py - for use in Chapter 10 PCfB\n",
    "# Read in each line of the example file, split it into \n",
    "# separate components, and write certain output to a separate file\n",
    "\n",
    "# Set the input file name\n",
    "# (The program must be run from within the directory \n",
    "#  that contains this data file)\n",
    "#InFileName = './logs/16process.out'\n",
    "InFileName = '/local/home/michelles/02-apcl-ddocent/18seq/logs/processP078.log'\n",
    "\n",
    "# Open the input file for reading\n",
    "InFile = open(InFileName, 'r')\n",
    "\n",
    "# Initialize the counter used to keep track of line numbers\n",
    "LineNumber = 0\n",
    "\n",
    "# Open the output file for writing\n",
    "# Do this *before* the loop, not inside it\n",
    "OutFileName=InFileName + \".tsv\"\n",
    "\n",
    "OutFile=open(OutFileName,'w') # You can append instead with 'a'\n",
    "\n",
    "# Loop through each line in the file\n",
    "for Line in InFile:\n",
    "        # Skip the header, line # 0\n",
    "        if LineNumber < 62:\n",
    "                if LineNumber > 13:\n",
    "                        # Remove the line ending characters\n",
    "                        Line=Line.strip('\\n')\n",
    "                        \n",
    "                        # Separate the line into a list of its tab-delimited components\n",
    "                        ElementList=Line.split('\\t')\n",
    "                        Barcode   = ElementList[0]\n",
    "                        Total_Reads = ElementList[1]\n",
    "                        Lack_RadTag = ElementList[2]  \n",
    "                        Low_Quality = ElementList[3]\n",
    "                        Retained = ElementList[4]\n",
    "                        \n",
    "                        # Use the % operator to generate a string \n",
    "                        # We can use this for output both to the screen and to a file\n",
    "                        OutputString = \"%s\\t%s\\t%s\\t%s\\t%s\" % (ElementList[0], ElementList[1], ElementList[2], ElementList[3], ElementList[4])\n",
    "                           \n",
    "                        # Can still print to the screen then write to a file\n",
    "                        print (OutputString)\n",
    "                        \n",
    "                        # Unlike print statements, .write needs a linefeed\n",
    "                        OutFile.write(OutputString+\"\\n\")  \n",
    "                \n",
    "        # Index the counter used to keep track of line numbers\n",
    "        LineNumber = LineNumber + 1\n",
    "        # After the loop is completed, close the files\n",
    "InFile.close()\n",
    "OutFile.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "barcodes         names_083.tsv        processP074.log      processP078.log\r\n",
      "index-seq18.tsv  names_084.tsv        processP074.log.tsv  processP078.log.tsv\r\n",
      "names_073.tsv    names_085.tsv        processP075.log      processP083.log\r\n",
      "names_074.tsv    names_086.tsv        processP075.log.tsv  processP084.log\r\n",
      "names_075.tsv    names_087.tsv        processP076.log      processP085.log\r\n",
      "names_076.tsv    names_088.tsv        processP076.log.tsv  processP086.log\r\n",
      "names_077.tsv    processP073.log      processP077.log      processP087.log\r\n",
      "names_078.tsv    processP073.log.tsv  processP077.log.tsv  processP088.log\r\n"
     ]
    }
   ],
   "source": [
    "%ls 02-apcl-ddocent/18seq/logs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AAACAC\t311328\t920\t3599\t305059\n",
      "AAACGA\t755744\t2186\t9135\t740309\n",
      "AAAGTC\t108927\t437\t1147\t106624\n",
      "AACGGT\t576085\t3022\t6703\t562778\n",
      "AACTTC\t315241\t974\t3780\t308809\n",
      "TGCTCA\t298717\t746\t3231\t292782\n",
      "AAGAAC\t131174\t374\t1437\t128658\n",
      "AATGTG\t455821\t1969\t5215\t446204\n",
      "ACATGT\t434612\t1397\t4899\t425774\n",
      "ACCAAA\t121636\t322\t1503\t119154\n",
      "ACGATA\t189919\t531\t2239\t186140\n",
      "ACGTTT\t152146\t808\t1869\t148551\n",
      "ACTAGG\t235599\t843\t2607\t230934\n",
      "ACTCCA\t254725\t822\t2984\t249448\n",
      "AGACTC\t288005\t1053\t3232\t282263\n",
      "AGCATT\t282\t22\t14\t241\n",
      "AGGAGA\t68544\t282\t749\t67136\n",
      "AGTAAG\t254476\t2358\t2755\t248113\n",
      "AGTCCT\t70027\t466\t783\t68428\n",
      "AGTTAC\t246702\t704\t2847\t241986\n",
      "ATAACC\t22180\t149\t304\t21574\n",
      "ATCGCA\t41478\t148\t479\t40639\n",
      "ATCTCG\t121920\t466\t1486\t119380\n",
      "ATGGAG\t265725\t4785\t2922\t256433\n",
      "ATGTCC\t87270\t246\t937\t85696\n",
      "CAATTG\t542588\t1969\t6129\t531431\n",
      "CAGAGT\t85432\t600\t896\t83436\n",
      "CATCAG\t155337\t916\t1741\t151833\n",
      "CATCTC\t338982\t1179\t3714\t332147\n",
      "CCACTT\t86070\t969\t1012\t83667\n",
      "CCCATA\t787720\t2645\t8874\t771082\n",
      "CCTGAA\t708\t7\t19\t681\n",
      "CGAAAC\t15040\t59\t149\t14756\n",
      "CGAATG\t82167\t270\t915\t80402\n",
      "GACGTT\t39678\t185\t478\t38773\n",
      "GATACA\t306129\t933\t3410\t300085\n",
      "GCAGAA\t8253\t36\t124\t8067\n",
      "GGGATA\t340583\t1309\t3839\t333588\n",
      "GGTGAA\t1043061\t4178\t11249\t1021601\n",
      "GTAGCT\t311462\t1653\t3505\t304543\n",
      "GTCTAT\t249166\t850\t2682\t244355\n",
      "GTTCAG\t329601\t1269\t3346\t323208\n",
      "TAAGAC\t275493\t1005\t3021\t270036\n",
      "TACCAG\t84210\t404\t976\t82387\n",
      "TCAATC\t224252\t1374\t2426\t219162\n",
      "TCCAAA\t1232833\t5109\t14343\t1206595\n",
      "TCTGCT\t109658\t572\t1098\t107388\n",
      "TCTTAG\t44921\t524\t504\t43661\n"
     ]
    }
   ],
   "source": [
    "# latlon_3.py - for use in Chapter 10 PCfB\n",
    "# Read in each line of the example file, split it into \n",
    "# separate components, and write certain output to a separate file\n",
    "\n",
    "# Set the input file name\n",
    "# (The program must be run from within the directory \n",
    "#  that contains this data file)\n",
    "#InFileName = './logs/16process.out'\n",
    "InFileName = '/local/home/michelles/02-apcl-ddocent/18seq/logs/processP083.log'\n",
    "\n",
    "# Open the input file for reading\n",
    "InFile = open(InFileName, 'r')\n",
    "\n",
    "# Initialize the counter used to keep track of line numbers\n",
    "LineNumber = 0\n",
    "\n",
    "# Open the output file for writing\n",
    "# Do this *before* the loop, not inside it\n",
    "OutFileName=InFileName + \".tsv\"\n",
    "\n",
    "OutFile=open(OutFileName,'w') # You can append instead with 'a'\n",
    "\n",
    "# Loop through each line in the file\n",
    "for Line in InFile:\n",
    "        # Skip the header, line # 0\n",
    "        if LineNumber < 62:\n",
    "                if LineNumber > 13:\n",
    "                        # Remove the line ending characters\n",
    "                        Line=Line.strip('\\n')\n",
    "                        \n",
    "                        # Separate the line into a list of its tab-delimited components\n",
    "                        ElementList=Line.split('\\t')\n",
    "                        Barcode   = ElementList[0]\n",
    "                        Total_Reads = ElementList[1]\n",
    "                        Lack_RadTag = ElementList[2]  \n",
    "                        Low_Quality = ElementList[3]\n",
    "                        Retained = ElementList[4]\n",
    "                        \n",
    "                        # Use the % operator to generate a string \n",
    "                        # We can use this for output both to the screen and to a file\n",
    "                        OutputString = \"%s\\t%s\\t%s\\t%s\\t%s\" % (ElementList[0], ElementList[1], ElementList[2], ElementList[3], ElementList[4])\n",
    "                           \n",
    "                        # Can still print to the screen then write to a file\n",
    "                        print (OutputString)\n",
    "                        \n",
    "                        # Unlike print statements, .write needs a linefeed\n",
    "                        OutFile.write(OutputString+\"\\n\")  \n",
    "                \n",
    "        # Index the counter used to keep track of line numbers\n",
    "        LineNumber = LineNumber + 1\n",
    "        # After the loop is completed, close the files\n",
    "InFile.close()\n",
    "OutFile.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AAACAC\t171222\t1255\t2515\t166476\n",
      "AAACGA\t398260\t1678\t5341\t388907\n",
      "AAAGTC\t395168\t1847\t4987\t386349\n",
      "AACGGT\t409633\t2208\t5723\t399318\n",
      "AACTTC\t184912\t1240\t2358\t180156\n",
      "TGCTCA\t343243\t1009\t4089\t336232\n",
      "AAGAAC\t928026\t4443\t11321\t907424\n",
      "AATGTG\t297306\t3924\t3646\t288199\n",
      "ACATGT\t634996\t2627\t8119\t620791\n",
      "ACCAAA\t775362\t2243\t9444\t759176\n",
      "ACGATA\t517074\t1646\t6794\t505722\n",
      "ACGTTT\t549371\t3730\t7517\t535000\n",
      "ACTAGG\t375243\t1457\t4367\t367363\n",
      "ACTCCA\t124858\t324\t1576\t122195\n",
      "AGACTC\t878889\t2423\t10835\t860529\n",
      "AGCATT\t804778\t4668\t10273\t785124\n",
      "AGGAGA\t779610\t2575\t9248\t763275\n",
      "AGTAAG\t412781\t1375\t5020\t404123\n",
      "AGTCCT\t835586\t5006\t10063\t815984\n",
      "AGTTAC\t83326\t249\t1110\t81518\n",
      "ATAACC\t368800\t1242\t4684\t360809\n",
      "ATCGCA\t1188357\t3219\t15169\t1162942\n",
      "ATCTCG\t733162\t2353\t9407\t717290\n",
      "ATGGAG\t932906\t3420\t11226\t912500\n",
      "ATGTCC\t767626\t2061\t9363\t751772\n",
      "CAATTG\t995208\t4223\t11814\t972528\n",
      "CAGAGT\t909543\t4402\t10219\t888318\n",
      "CATCAG\t354015\t7559\t4244\t340245\n",
      "CATCTC\t457972\t1630\t5757\t448020\n",
      "CCACTT\t865666\t3149\t11018\t846508\n",
      "CCCATA\t697079\t2504\t8591\t681476\n",
      "CCTGAA\t685316\t2529\t8002\t670798\n",
      "CGAAAC\t287923\t898\t3503\t281786\n",
      "CGAATG\t1034361\t4061\t12007\t1011806\n",
      "GACGTT\t516561\t2186\t6574\t504907\n",
      "GATACA\t730757\t2204\t8783\t715825\n",
      "GCAGAA\t1366014\t4730\t15336\t1338553\n",
      "GGGATA\t649121\t2082\t7869\t635496\n",
      "GGTGAA\t401201\t1406\t4713\t392591\n",
      "GTAGCT\t1105107\t3958\t13117\t1081663\n",
      "GTCTAT\t833500\t2784\t9704\t816092\n",
      "GTTCAG\t75281\t325\t902\t73635\n",
      "TAAGAC\t817378\t2818\t9550\t800371\n",
      "TACCAG\t351923\t1107\t4311\t344530\n",
      "TCAATC\t604654\t2141\t7044\t592049\n",
      "TCCAAA\t745959\t2239\t9211\t730102\n",
      "TCTGCT\t489962\t1507\t5970\t479335\n",
      "TCTTAG\t303611\t2694\t3577\t295813\n"
     ]
    }
   ],
   "source": [
    "# latlon_3.py - for use in Chapter 10 PCfB\n",
    "# Read in each line of the example file, split it into \n",
    "# separate components, and write certain output to a separate file\n",
    "\n",
    "# Set the input file name\n",
    "# (The program must be run from within the directory \n",
    "#  that contains this data file)\n",
    "#InFileName = './logs/16process.out'\n",
    "InFileName = '/local/home/michelles/02-apcl-ddocent/18seq/logs/processP084.log'\n",
    "\n",
    "# Open the input file for reading\n",
    "InFile = open(InFileName, 'r')\n",
    "\n",
    "# Initialize the counter used to keep track of line numbers\n",
    "LineNumber = 0\n",
    "\n",
    "# Open the output file for writing\n",
    "# Do this *before* the loop, not inside it\n",
    "OutFileName=InFileName + \".tsv\"\n",
    "\n",
    "OutFile=open(OutFileName,'w') # You can append instead with 'a'\n",
    "\n",
    "# Loop through each line in the file\n",
    "for Line in InFile:\n",
    "        # Skip the header, line # 0\n",
    "        if LineNumber < 62:\n",
    "                if LineNumber > 13:\n",
    "                        # Remove the line ending characters\n",
    "                        Line=Line.strip('\\n')\n",
    "                        \n",
    "                        # Separate the line into a list of its tab-delimited components\n",
    "                        ElementList=Line.split('\\t')\n",
    "                        Barcode   = ElementList[0]\n",
    "                        Total_Reads = ElementList[1]\n",
    "                        Lack_RadTag = ElementList[2]  \n",
    "                        Low_Quality = ElementList[3]\n",
    "                        Retained = ElementList[4]\n",
    "                        \n",
    "                        # Use the % operator to generate a string \n",
    "                        # We can use this for output both to the screen and to a file\n",
    "                        OutputString = \"%s\\t%s\\t%s\\t%s\\t%s\" % (ElementList[0], ElementList[1], ElementList[2], ElementList[3], ElementList[4])\n",
    "                           \n",
    "                        # Can still print to the screen then write to a file\n",
    "                        print (OutputString)\n",
    "                        \n",
    "                        # Unlike print statements, .write needs a linefeed\n",
    "                        OutFile.write(OutputString+\"\\n\")  \n",
    "                \n",
    "        # Index the counter used to keep track of line numbers\n",
    "        LineNumber = LineNumber + 1\n",
    "        # After the loop is completed, close the files\n",
    "InFile.close()\n",
    "OutFile.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AAACAC\t169523\t892\t2425\t165343\n",
      "AAACGA\t630419\t2271\t9686\t615165\n",
      "AAAGTC\t510347\t2368\t7645\t497779\n",
      "AACGGT\t656349\t2924\t10338\t639481\n",
      "AACTTC\t560307\t2034\t9115\t546296\n",
      "TGCTCA\t648183\t2407\t9057\t633478\n",
      "AAGAAC\t332799\t1219\t4685\t325148\n",
      "AATGTG\t601870\t3774\t8714\t586468\n",
      "ACATGT\t195918\t1182\t3079\t190684\n",
      "ACCAAA\t141647\t640\t2371\t137807\n",
      "ACGATA\t388658\t1336\t6026\t379233\n",
      "ACGTTT\t238006\t2517\t3988\t230225\n",
      "ACTAGG\t162425\t873\t2284\t158396\n",
      "ACTCCA\t202130\t879\t3293\t196889\n",
      "AGACTC\t168312\t662\t2675\t164048\n",
      "AGCATT\t166451\t1947\t2646\t160912\n",
      "AGGAGA\t252414\t1483\t3810\t245687\n",
      "AGTAAG\t267975\t2145\t3897\t260452\n",
      "AGTCCT\t310973\t3049\t4651\t301925\n",
      "AGTTAC\t361302\t1936\t5701\t352023\n",
      "ATAACC\t187006\t1410\t3057\t181570\n",
      "ATCGCA\t406333\t1485\t6304\t396541\n",
      "ATCTCG\t401283\t1414\t6536\t391385\n",
      "ATGGAG\t294887\t1468\t5141\t286691\n",
      "ATGTCC\t442454\t1641\t6372\t432034\n",
      "CAATTG\t388625\t2359\t5605\t378442\n",
      "CAGAGT\t290174\t2607\t4213\t281729\n",
      "CATCAG\t244122\t2107\t3714\t236932\n",
      "CATCTC\t351344\t1140\t5012\t343193\n",
      "CCACTT\t318469\t1063\t4604\t311355\n",
      "CCCATA\t300352\t988\t4623\t293237\n",
      "CCTGAA\t304803\t1258\t4531\t297370\n",
      "CGAAAC\t511091\t2021\t7284\t498957\n",
      "CGAATG\t301571\t2599\t4442\t292915\n",
      "GACGTT\t339191\t1590\t4954\t330825\n",
      "GATACA\t424956\t2086\t6427\t414444\n",
      "GCAGAA\t365673\t1917\t5047\t356935\n",
      "GGGATA\t290287\t1477\t4408\t283059\n",
      "GGTGAA\t194731\t720\t2546\t190519\n",
      "GTAGCT\t225050\t979\t3322\t219509\n",
      "GTCTAT\t512641\t2021\t7280\t500764\n",
      "GTTCAG\t443673\t1881\t5969\t433457\n",
      "TAAGAC\t292436\t1419\t4200\t285196\n",
      "TACCAG\t336684\t1291\t5083\t328714\n",
      "TCAATC\t415087\t1360\t5547\t406048\n",
      "TCCAAA\t375762\t1555\t5878\t366567\n",
      "TCTGCT\t342143\t1586\t5071\t333805\n",
      "TCTTAG\t197464\t1615\t2716\t192201\n"
     ]
    }
   ],
   "source": [
    "# latlon_3.py - for use in Chapter 10 PCfB\n",
    "# Read in each line of the example file, split it into \n",
    "# separate components, and write certain output to a separate file\n",
    "\n",
    "# Set the input file name\n",
    "# (The program must be run from within the directory \n",
    "#  that contains this data file)\n",
    "#InFileName = './logs/16process.out'\n",
    "InFileName = '/local/home/michelles/02-apcl-ddocent/18seq/logs/processP085.log'\n",
    "\n",
    "# Open the input file for reading\n",
    "InFile = open(InFileName, 'r')\n",
    "\n",
    "# Initialize the counter used to keep track of line numbers\n",
    "LineNumber = 0\n",
    "\n",
    "# Open the output file for writing\n",
    "# Do this *before* the loop, not inside it\n",
    "OutFileName=InFileName + \".tsv\"\n",
    "\n",
    "OutFile=open(OutFileName,'w') # You can append instead with 'a'\n",
    "\n",
    "# Loop through each line in the file\n",
    "for Line in InFile:\n",
    "        # Skip the header, line # 0\n",
    "        if LineNumber < 62:\n",
    "                if LineNumber > 13:\n",
    "                        # Remove the line ending characters\n",
    "                        Line=Line.strip('\\n')\n",
    "                        \n",
    "                        # Separate the line into a list of its tab-delimited components\n",
    "                        ElementList=Line.split('\\t')\n",
    "                        Barcode   = ElementList[0]\n",
    "                        Total_Reads = ElementList[1]\n",
    "                        Lack_RadTag = ElementList[2]  \n",
    "                        Low_Quality = ElementList[3]\n",
    "                        Retained = ElementList[4]\n",
    "                        \n",
    "                        # Use the % operator to generate a string \n",
    "                        # We can use this for output both to the screen and to a file\n",
    "                        OutputString = \"%s\\t%s\\t%s\\t%s\\t%s\" % (ElementList[0], ElementList[1], ElementList[2], ElementList[3], ElementList[4])\n",
    "                           \n",
    "                        # Can still print to the screen then write to a file\n",
    "                        print (OutputString)\n",
    "                        \n",
    "                        # Unlike print statements, .write needs a linefeed\n",
    "                        OutFile.write(OutputString+\"\\n\")  \n",
    "                \n",
    "        # Index the counter used to keep track of line numbers\n",
    "        LineNumber = LineNumber + 1\n",
    "        # After the loop is completed, close the files\n",
    "InFile.close()\n",
    "OutFile.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AAACAC\t67500\t480\t832\t65833\n",
      "AAACGA\t101310\t583\t1531\t98588\n",
      "AAAGTC\t94273\t484\t1254\t92038\n",
      "AACGGT\t295794\t1726\t4350\t287960\n",
      "AACTTC\t338254\t1047\t4962\t330313\n",
      "TGCTCA\t282192\t914\t3817\t275864\n",
      "AAGAAC\t220814\t1013\t2905\t215614\n",
      "AATGTG\t177739\t1179\t2395\t173333\n",
      "ACATGT\t294444\t1268\t3859\t287603\n",
      "ACCAAA\t190400\t714\t2666\t186009\n",
      "ACGATA\t78723\t448\t1178\t76598\n",
      "ACGTTT\t159099\t1398\t2283\t154615\n",
      "ACTAGG\t147953\t882\t1986\t144226\n",
      "ACTCCA\t247040\t756\t3365\t241745\n",
      "AGACTC\t203438\t650\t2804\t198862\n",
      "AGCATT\t108481\t880\t1538\t105469\n",
      "AGGAGA\t409995\t1552\t5436\t400613\n",
      "AGTAAG\t216719\t921\t2895\t211657\n",
      "AGTCCT\t91775\t989\t1234\t89045\n",
      "AGTTAC\t117727\t329\t1573\t115182\n",
      "ATAACC\t104174\t348\t1418\t101852\n",
      "ATCGCA\t302140\t810\t4231\t295553\n",
      "ATCTCG\t271613\t1066\t3852\t265375\n",
      "ATGGAG\t225614\t981\t2999\t220182\n",
      "ATGTCC\t211692\t768\t2876\t207178\n",
      "CAATTG\t299\t44\t14\t239\n",
      "CAGAGT\t167160\t2042\t2426\t161713\n",
      "CATCAG\t175319\t1450\t2272\t170523\n",
      "CATCTC\t231471\t855\t3177\t226411\n",
      "CCACTT\t186168\t1216\t2707\t181297\n",
      "CCCATA\t349739\t1353\t4752\t341689\n",
      "CCTGAA\t134806\t689\t1852\t131515\n",
      "CGAAAC\t147974\t654\t2122\t144286\n",
      "CGAATG\t271604\t1410\t3535\t265210\n",
      "GACGTT\t219107\t791\t2944\t213952\n",
      "GATACA\t117764\t508\t1569\t115006\n",
      "GCAGAA\t173596\t658\t2148\t169844\n",
      "GGGATA\t266573\t1193\t3564\t260379\n",
      "GGTGAA\t205835\t919\t2726\t200895\n",
      "GTAGCT\t103417\t694\t1414\t100767\n",
      "GTCTAT\t351033\t1588\t4600\t342898\n",
      "GTTCAG\t322141\t1740\t4075\t314528\n",
      "TAAGAC\t102404\t892\t1596\t99375\n",
      "TACCAG\t202046\t1154\t2904\t196812\n",
      "TCAATC\t210216\t1122\t2856\t205223\n",
      "TCCAAA\t158105\t588\t2357\t154286\n",
      "TCTGCT\t50781\t315\t787\t49378\n",
      "TCTTAG\t147931\t866\t2062\t144266\n"
     ]
    }
   ],
   "source": [
    "# latlon_3.py - for use in Chapter 10 PCfB\n",
    "# Read in each line of the example file, split it into \n",
    "# separate components, and write certain output to a separate file\n",
    "\n",
    "# Set the input file name\n",
    "# (The program must be run from within the directory \n",
    "#  that contains this data file)\n",
    "#InFileName = './logs/16process.out'\n",
    "InFileName = '/local/home/michelles/02-apcl-ddocent/18seq/logs/processP086.log'\n",
    "\n",
    "# Open the input file for reading\n",
    "InFile = open(InFileName, 'r')\n",
    "\n",
    "# Initialize the counter used to keep track of line numbers\n",
    "LineNumber = 0\n",
    "\n",
    "# Open the output file for writing\n",
    "# Do this *before* the loop, not inside it\n",
    "OutFileName=InFileName + \".tsv\"\n",
    "\n",
    "OutFile=open(OutFileName,'w') # You can append instead with 'a'\n",
    "\n",
    "# Loop through each line in the file\n",
    "for Line in InFile:\n",
    "        # Skip the header, line # 0\n",
    "        if LineNumber < 62:\n",
    "                if LineNumber > 13:\n",
    "                        # Remove the line ending characters\n",
    "                        Line=Line.strip('\\n')\n",
    "                        \n",
    "                        # Separate the line into a list of its tab-delimited components\n",
    "                        ElementList=Line.split('\\t')\n",
    "                        Barcode   = ElementList[0]\n",
    "                        Total_Reads = ElementList[1]\n",
    "                        Lack_RadTag = ElementList[2]  \n",
    "                        Low_Quality = ElementList[3]\n",
    "                        Retained = ElementList[4]\n",
    "                        \n",
    "                        # Use the % operator to generate a string \n",
    "                        # We can use this for output both to the screen and to a file\n",
    "                        OutputString = \"%s\\t%s\\t%s\\t%s\\t%s\" % (ElementList[0], ElementList[1], ElementList[2], ElementList[3], ElementList[4])\n",
    "                           \n",
    "                        # Can still print to the screen then write to a file\n",
    "                        print (OutputString)\n",
    "                        \n",
    "                        # Unlike print statements, .write needs a linefeed\n",
    "                        OutFile.write(OutputString+\"\\n\")  \n",
    "                \n",
    "        # Index the counter used to keep track of line numbers\n",
    "        LineNumber = LineNumber + 1\n",
    "        # After the loop is completed, close the files\n",
    "InFile.close()\n",
    "OutFile.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AAACAC\t190302\t4852\t3087\t181384\n",
      "AAACGA\t162175\t779\t2637\t158008\n",
      "AAAGTC\t152811\t557\t2153\t149325\n",
      "AACGGT\t303513\t1388\t4942\t295437\n",
      "AACTTC\t323025\t1495\t5229\t314543\n",
      "TGCTCA\t579153\t1997\t8812\t564999\n",
      "AAGAAC\t231566\t1316\t3525\t225397\n",
      "AATGTG\t486478\t4217\t8083\t471789\n",
      "ACATGT\t437502\t2459\t6876\t426044\n",
      "ACCAAA\t506837\t2133\t8208\t493714\n",
      "ACGATA\t742900\t2938\t11915\t724097\n",
      "ACGTTT\t313593\t3827\t6021\t302150\n",
      "ACTAGG\t378374\t3473\t6092\t366789\n",
      "ACTCCA\t424307\t1837\t6839\t413085\n",
      "AGACTC\t495279\t1923\t8148\t482568\n",
      "AGCATT\t421086\t2758\t6709\t408887\n",
      "AGGAGA\t392074\t1560\t5816\t382316\n",
      "AGTAAG\t242603\t1238\t3746\t236281\n",
      "AGTCCT\t239804\t2258\t3668\t232625\n",
      "AGTTAC\t264139\t1263\t4251\t257314\n",
      "ATAACC\t405359\t1730\t6311\t394905\n",
      "ATCGCA\t240659\t925\t3749\t234796\n",
      "ATCTCG\t498821\t2857\t8550\t484865\n",
      "ATGGAG\t260688\t3463\t4523\t251282\n",
      "ATGTCC\t337039\t1808\t5442\t327983\n",
      "CAATTG\t156422\t3082\t2667\t149931\n",
      "CAGAGT\t191687\t2604\t2990\t184951\n",
      "CATCAG\t266339\t2834\t4342\t257734\n",
      "CATCTC\t487292\t2608\t8034\t474112\n",
      "CCACTT\t413108\t1505\t7116\t402784\n",
      "CCCATA\t781551\t2864\t12515\t761741\n",
      "CCTGAA\t444845\t2741\t6876\t432787\n",
      "CGAAAC\t687481\t4032\t11164\t668730\n",
      "CGAATG\t198403\t2536\t3351\t191292\n",
      "GACGTT\t197673\t1601\t3379\t191597\n",
      "GATACA\t149977\t848\t2347\t145969\n",
      "GCAGAA\t277477\t1181\t4304\t270382\n",
      "GGGATA\t420971\t2176\t6752\t409570\n",
      "GGTGAA\t313520\t2169\t5003\t304564\n",
      "GTAGCT\t122099\t548\t1828\t119152\n",
      "GTCTAT\t283639\t1709\t4321\t276060\n",
      "GTTCAG\t277394\t1484\t3967\t270260\n",
      "TAAGAC\t601218\t2719\t8906\t586707\n",
      "TACCAG\t265832\t1651\t4364\t258425\n",
      "TCAATC\t546003\t1915\t7848\t533008\n",
      "TCCAAA\t515924\t1944\t7916\t503124\n",
      "TCTGCT\t472544\t2034\t7256\t460467\n",
      "TCTTAG\t321653\t1496\t4875\t313582\n"
     ]
    }
   ],
   "source": [
    "# latlon_3.py - for use in Chapter 10 PCfB\n",
    "# Read in each line of the example file, split it into \n",
    "# separate components, and write certain output to a separate file\n",
    "\n",
    "# Set the input file name\n",
    "# (The program must be run from within the directory \n",
    "#  that contains this data file)\n",
    "#InFileName = './logs/16process.out'\n",
    "InFileName = '/local/home/michelles/02-apcl-ddocent/18seq/logs/processP087.log'\n",
    "\n",
    "# Open the input file for reading\n",
    "InFile = open(InFileName, 'r')\n",
    "\n",
    "# Initialize the counter used to keep track of line numbers\n",
    "LineNumber = 0\n",
    "\n",
    "# Open the output file for writing\n",
    "# Do this *before* the loop, not inside it\n",
    "OutFileName=InFileName + \".tsv\"\n",
    "\n",
    "OutFile=open(OutFileName,'w') # You can append instead with 'a'\n",
    "\n",
    "# Loop through each line in the file\n",
    "for Line in InFile:\n",
    "        # Skip the header, line # 0\n",
    "        if LineNumber < 62:\n",
    "                if LineNumber > 13:\n",
    "                        # Remove the line ending characters\n",
    "                        Line=Line.strip('\\n')\n",
    "                        \n",
    "                        # Separate the line into a list of its tab-delimited components\n",
    "                        ElementList=Line.split('\\t')\n",
    "                        Barcode   = ElementList[0]\n",
    "                        Total_Reads = ElementList[1]\n",
    "                        Lack_RadTag = ElementList[2]  \n",
    "                        Low_Quality = ElementList[3]\n",
    "                        Retained = ElementList[4]\n",
    "                        \n",
    "                        # Use the % operator to generate a string \n",
    "                        # We can use this for output both to the screen and to a file\n",
    "                        OutputString = \"%s\\t%s\\t%s\\t%s\\t%s\" % (ElementList[0], ElementList[1], ElementList[2], ElementList[3], ElementList[4])\n",
    "                           \n",
    "                        # Can still print to the screen then write to a file\n",
    "                        print (OutputString)\n",
    "                        \n",
    "                        # Unlike print statements, .write needs a linefeed\n",
    "                        OutFile.write(OutputString+\"\\n\")  \n",
    "                \n",
    "        # Index the counter used to keep track of line numbers\n",
    "        LineNumber = LineNumber + 1\n",
    "        # After the loop is completed, close the files\n",
    "InFile.close()\n",
    "OutFile.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AAACAC\t101429\t3063\t1398\t96505\n",
      "AAACGA\t105575\t465\t1529\t103032\n",
      "AAAGTC\t442092\t2087\t6428\t431190\n",
      "AACGGT\t228041\t1309\t3320\t222145\n",
      "AACTTC\t342331\t1470\t5210\t334067\n",
      "TGCTCA\t279245\t1395\t3953\t272535\n",
      "AAGAAC\t141989\t929\t2012\t138294\n",
      "AATGTG\t167020\t1280\t2458\t162488\n",
      "ACATGT\t448364\t2574\t6984\t436488\n",
      "ACCAAA\t401860\t1276\t6099\t392454\n",
      "ACGATA\t208146\t1162\t3217\t202821\n",
      "ACGTTT\t311980\t2483\t4976\t302957\n",
      "ACTAGG\t330333\t2110\t5081\t321535\n",
      "ACTCCA\t438927\t1409\t6405\t428998\n",
      "AGACTC\t322891\t1089\t4624\t315416\n",
      "AGCATT\t388728\t2192\t6004\t378534\n",
      "AGGAGA\t430482\t1775\t6179\t420202\n",
      "AGTAAG\t417868\t2866\t6758\t406039\n",
      "AGTCCT\t370447\t2513\t5600\t360585\n",
      "AGTTAC\t313094\t1820\t4776\t304896\n",
      "ATAACC\t165242\t730\t2515\t161155\n",
      "ATCGCA\t277391\t972\t4175\t270977\n",
      "ATCTCG\t86183\t545\t1262\t84017\n",
      "ATGGAG\t65111\t374\t946\t63484\n",
      "ATGTCC\t115700\t291\t1612\t113268\n",
      "CAATTG\t101761\t588\t1396\t99310\n",
      "CAGAGT\t101876\t1237\t1443\t98643\n",
      "CATCAG\t352997\t1642\t4847\t344524\n",
      "CATCTC\t656117\t2118\t9680\t640911\n",
      "CCACTT\t360644\t1280\t5327\t352047\n",
      "CCCATA\t497098\t1460\t7151\t485691\n",
      "CCTGAA\t691\t45\t35\t606\n",
      "CGAAAC\t232062\t1131\t3257\t226082\n",
      "CGAATG\t143154\t803\t1971\t139532\n",
      "GACGTT\t192197\t966\t2770\t187446\n",
      "GATACA\t235463\t798\t3434\t230024\n",
      "GCAGAA\t189929\t761\t2543\t185632\n",
      "GGGATA\t192784\t674\t2809\t188314\n",
      "GGTGAA\t182358\t655\t2464\t178187\n",
      "GTAGCT\t176624\t728\t2439\t172632\n",
      "GTCTAT\t347090\t1446\t4745\t338944\n",
      "GTTCAG\t149682\t583\t2061\t146225\n",
      "TAAGAC\t83104\t399\t1117\t81162\n",
      "TACCAG\t60126\t250\t897\t58690\n",
      "TCAATC\t221142\t669\t3019\t216423\n",
      "TCCAAA\t231718\t758\t3214\t226586\n",
      "TCTGCT\t381881\t1269\t5473\t373330\n",
      "TCTTAG\t174390\t628\t2170\t170667\n"
     ]
    }
   ],
   "source": [
    "# latlon_3.py - for use in Chapter 10 PCfB\n",
    "# Read in each line of the example file, split it into \n",
    "# separate components, and write certain output to a separate file\n",
    "\n",
    "# Set the input file name\n",
    "# (The program must be run from within the directory \n",
    "#  that contains this data file)\n",
    "#InFileName = './logs/16process.out'\n",
    "InFileName = '/local/home/michelles/02-apcl-ddocent/18seq/logs/processP088.log'\n",
    "\n",
    "# Open the input file for reading\n",
    "InFile = open(InFileName, 'r')\n",
    "\n",
    "# Initialize the counter used to keep track of line numbers\n",
    "LineNumber = 0\n",
    "\n",
    "# Open the output file for writing\n",
    "# Do this *before* the loop, not inside it\n",
    "OutFileName=InFileName + \".tsv\"\n",
    "\n",
    "OutFile=open(OutFileName,'w') # You can append instead with 'a'\n",
    "\n",
    "# Loop through each line in the file\n",
    "for Line in InFile:\n",
    "        # Skip the header, line # 0\n",
    "        if LineNumber < 62:\n",
    "                if LineNumber > 13:\n",
    "                        # Remove the line ending characters\n",
    "                        Line=Line.strip('\\n')\n",
    "                        \n",
    "                        # Separate the line into a list of its tab-delimited components\n",
    "                        ElementList=Line.split('\\t')\n",
    "                        Barcode   = ElementList[0]\n",
    "                        Total_Reads = ElementList[1]\n",
    "                        Lack_RadTag = ElementList[2]  \n",
    "                        Low_Quality = ElementList[3]\n",
    "                        Retained = ElementList[4]\n",
    "                        \n",
    "                        # Use the % operator to generate a string \n",
    "                        # We can use this for output both to the screen and to a file\n",
    "                        OutputString = \"%s\\t%s\\t%s\\t%s\\t%s\" % (ElementList[0], ElementList[1], ElementList[2], ElementList[3], ElementList[4])\n",
    "                           \n",
    "                        # Can still print to the screen then write to a file\n",
    "                        print (OutputString)\n",
    "                        \n",
    "                        # Unlike print statements, .write needs a linefeed\n",
    "                        OutFile.write(OutputString+\"\\n\")  \n",
    "                \n",
    "        # Index the counter used to keep track of line numbers\n",
    "        LineNumber = LineNumber + 1\n",
    "        # After the loop is completed, close the files\n",
    "InFile.close()\n",
    "OutFile.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Move the newly created tsvs to the laptop with fetch and import them into the database "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Rename the sample files - used script in seq_proc.Rmd to generate mass command line text stored in SEQXX_all_rename.sh - this caused some problems because the copy and paste was off and it ended up deleting the APCL-analysis folder.  Because of this I am changing the first cd to a full path "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /local/home/michelles/02-apcl-ddocent/18seq/\n",
    "cd 01Pool/\n",
    "sh rename.for.dDocent_se_gz ../logs/names_073.tsv\n",
    "mv APCL* ../samples/\n",
    "cd ../02Pool/\n",
    "sh rename.for.dDocent_se_gz ../logs/names_074.tsv\n",
    "mv APCL* ../samples/\n",
    "cd ../03Pool/\n",
    "sh rename.for.dDocent_se_gz ../logs/names_075.tsv\n",
    "mv APCL* ../samples/\n",
    "cd ../04Pool/\n",
    "sh rename.for.dDocent_se_gz ../logs/names_076.tsv\n",
    "mv APCL* ../samples/\n",
    "cd ../05Pool/\n",
    "sh rename.for.dDocent_se_gz ../logs/names_077.tsv\n",
    "mv APCL* ../samples/\n",
    "cd ../06Pool/\n",
    "sh rename.for.dDocent_se_gz ../logs/names_078.tsv\n",
    "mv APCL* ../samples/\n",
    "cd ../07Pool/\n",
    "sh rename.for.dDocent_se_gz ../logs/names_083.tsv\n",
    "mv APCL* ../samples/\n",
    "cd ../08Pool/\n",
    "sh rename.for.dDocent_se_gz ../logs/names_084.tsv\n",
    "mv APCL* ../samples/\n",
    "cd ../09Pool/\n",
    "sh rename.for.dDocent_se_gz ../logs/names_085.tsv\n",
    "mv APCL* ../samples/\n",
    "cd ../10Pool/\n",
    "sh rename.for.dDocent_se_gz ../logs/names_086.tsv\n",
    "mv APCL* ../samples/\n",
    "cd ../11Pool/\n",
    "sh rename.for.dDocent_se_gz ../logs/names_087.tsv\n",
    "mv APCL* ../samples/\n",
    "cd ../12Pool/\n",
    "sh rename.for.dDocent_se_gz ../logs/names_088.tsv\n",
    "mv APCL* ../samples/\n",
    "cd ..\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Once all of the samples have been moved, move the gz files to the logs and then you can delete the pool directories"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "P074.fastq.gz\r\n"
     ]
    }
   ],
   "source": [
    "%rm -r *Pool"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## make a new directory for the current analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%mkdir ~/02-apcl-ddocent/APCL_analysis/21-03seq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Copy reference.fasta over from jonsfiles"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "% cp /local/home/michelles/02-apcl-ddocent/jonsfiles/reference.fasta /local/home/michelles/02-apcl-ddocent/APCL_analysis/test_28_03_seq/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## If you are running a small batch of samples instead of all samples ever collected, trim and map reads - started at 7:25am and finished at 9:45am on same day\n",
    "for this phase, dDocent will ask questions.\n",
    "* Are the number of samples correct?\n",
    "yes\n",
    "* Maximum number of processors to use for this analysis\n",
    "This depends on how many people are trying to use amphiprion right now.  The trim and map section isn't too intensive so it should be ok to use alot.  I used 30 for SEQ18 - 576 samples\n",
    "30\n",
    "* Maximum memory\n",
    "Again, consider who else is using the machine, for SEQ18 - 576 samples I used 150\n",
    "150\n",
    "* Quality trim?\n",
    "yes\n",
    "* Perform assembly?\n",
    "no - this is for creating the reference originally\n",
    "* Map reads?\n",
    "yes\n",
    "* Adjust default parameters \n",
    "default match score is 1, default mismatch is 4, gap penalty is 6 - I used defaults for all\n",
    "* call SNPs\n",
    "no\n",
    "* enter email address to be notified when it is done running"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "dDocent"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  }
 ],
 "metadata": {
  "celltoolbar": "Raw Cell Format",
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
