---
title: "Sequencing Prep ddRADSeq"
output: html_notebook
params:
  date: '2018-09-13'
  facility: PrincetonLSI
  first_baits: B010
  frag_length: 430
  last_baits: B012
  library_type: ddRADSeq
  num_lanes: 2
  pe_or_se: se
  platform: hiseq
  seq: SEQ28
  database: FALSE
---
```{r setup, include=FALSE}
knitr::opts_chunk$set(include = FALSE)
library(tidyverse)
library(knitr)
library(kableExtra)

database = params$database
```

```{r eval=database}
source("~/db-connections.R")
lab <- read_db("Laboratory")
```


```{r eval=database}
# pull in the baits that are going to be pooled for sequencing
baits <- lab %>% 
  tbl("baits") %>% 
  collect() 

# update the db to reflect pooling
change <- baits %>% 
  filter(baits_id >= params$first_baits & baits_id <= params$last_baits) %>% 
  mutate(seq = params$seq)

baits <- change_rows(baits, change, "baits_id")
dbWriteTable(lab, "baits", baits, row.names=F, overwrite = T)
```

```{r eval=!database}
change <- tribble(~baits_id, ~ng_ul, ~date, ~seq,	
  "B003",0.768,"2018-09-13","SEQ28",
	"B002",0.636,"2018-09-13","SEQ28",
	"B001",0.744,"2018-09-13","SEQ28",
	"B004",0.2,"2018-09-12","SEQ29",
	"B005",0.754,"2018-09-12","SEQ29",
	"B006",0.344,"2018-09-12","SEQ29",
	"B007",1.34,"2018-09-12","SEQ30",
	"B008",1.466,"2018-09-12","SEQ30",
	"B009",0.844,"2018-09-12","SEQ30",
	"B010",2.14,"2018-09-12","SEQ31",
	"B011",1.788,"2018-09-12","SEQ31",
	"B012",3.42,"2018-09-12","SEQ31") %>% 
  filter(seq == params$seq)
```


```{r}
# calculate the percent concentration of DNA in each bait
change <- change %>% 
  mutate(vol = 20, 
    perc = as.numeric(ng_ul)/as.numeric(vol))

# calculate the percent concentration of the pooled baits
tot <- change %>% 
   # sum of percents
  summarise(tot_perc = sum(perc)) %>%
  # sum of volumes
  mutate(vol = change$vol[1] * 3,  
    # calculate the ng_ul of the combined DNA
    ng_ul_lib = tot_perc * vol, 
    # calculate the nM of the combined DNA (430bp fragment length)
    nM_lib = (ng_ul_lib/(660*params$frag_length))*10^6, 
    # calculate the volume of library to produce 10nM
    vol_lib_ul = (30*10)/nM_lib) # 30 because that is our desired final volume to send to seq, 10 because we want to send 10nM 
```
```{r echo=FALSE}
knitr::kable(tot) %>% 
  kableExtra::kable_styling("bordered") %>% 
  add_header_above(c("vol_lib_ul tells us how much library to combine with water to send out" = 5))
```
```{r eval=database}
# add to seq table in the db
# dat_db <- lab %>% 
#   tbl("sequencing") %>% 
#   collect()

# create a row to add to the seq table in the db
# change <- tibble(seq_id = params$seq, date = params$date, library_type = params$library_type, facility = params$facility, platform = params$platform, num_lanes = params$num_lanes, pe_se = params$pe_or_se, vol_pooled_libs = tot$seq_vol, vol_TE = (30-tot$seq_vol), contents = paste(params$first_baits, params$last_baits, sep = "-"))

# dat_db <- rbind(dat_db, change) %>% 
#   mutate(vol_pooled_libs = formatC(vol_pooled_libs, width = 1), 
#     vol_TE = formatC(vol_TE, width = 1))

# dbWriteTable(lab, "sequencing", change, row.names=F, overwrite = F, append = T)
```

Label a vial with the SEQ number on top (SEQ 04) and  SEQ # date nM ng/µL indices on the side for example (this really does fit):

Pinsky
SEQ 04
12-10-2014
10 nM 
Indices 4,5,6,7

Stuff a kim wipe into a 50mL falcon tube, place the diluted 10nM sequencing tube inside, stuff in another kim wipe, screw on the cap and place in a styrofoam cooler with an icepack.  We want to keep the sample cool but we do not want to freeze it.  If it isn’t ready to depart immediately, keep the 50mL falcon with the sequencing tube in it in the fridge.

Place the original pooled sequencing sample vial in the freezer box in the -20˚C freezer

##Update 4/13/2015 - MRS
It is possible to calculate the concentration of each pool and then add 100ng of each:

Based on the qubit quantification (X), determine the volume of sample (Y) that will supply 100ng to the sequencing pool:

100ng ÷ X ng/µL = Y µL

The concentration of this pooled sequencing sample is calculated in the next cell by dividing 400 by the sum volume (400 because you added 100ng of 4 pools) - this value should calculate automatically in the spreadsheet

We are using 430bp fragments (375 from pippin plus 55 added by illumina primer for index and flowcell attachment) - enter this value into the spreadsheet

The spreadsheet will calculate nmol/µL by dividing the concentration by the product of 660 and the number of basepairs (430) …660*430=283,800

The spreadsheet will calculate nM (aka nmol/L) by multiplying nmol/µL by one million




Calculate the 4 pools independently on the PCR sheet in the Sample_Data file

Sum the volume on the Sequencing sheet in the Sample_Data file

Combine the 4 pools based on the volume calculated 


The next two cells in the Sequencing spreadsheet calculate the volume of sample and the volume of water to be combined in this new tube  - the volume of pooled sequencing sample is calculated by dividing 300 by the nM and the volume of water is calculated by subtracting the volume of sequencing sample from 30.

Combine the sequencing sample and Pure water in the new tube.  

Stuff a kim wipe into a 50mL falcon tube, place the diluted 10nM sequencing tube inside, stuff in another kim wipe, screw on the cap and place in a styrofoam cooler with an icepack.  We want to keep the sample cool but we do not want to freeze it.  If it isn’t ready to depart immediately, keep the 50mL falcon with the sequencing tube in it in the fridge.

Place the original pooled sequencing sample vial in the freezer box in the -20

If sending to Princeton, don’t generate the PO until the seq is complete (particularly if you are submitting more than one seq within a month).  Ask Princeton if they can add some description to the quote to make it look different than the rest of the quotes they have sent (like add the seq number).  We have had problems in the past with multiple quotes from similar dates with no other difference on the page.  Had to add a line of text saying “This invoice is for sequencing done to APCL12 DNA on February 6th, 2016”.

If sending the sample to Princeton, Malin will fill out the submission form on this webpage:
https://htseq.princeton.edu

Princeton also has guidelines on this webpage:
http://www.princeton.edu/genomics/sequencing/instructions/sample-preparation/

Their guidelines differ from our personal contact’s recommendations at Princeton and we will follow our own set of guidelines unless instructed otherwise by Malin.









