---
title: "Create a budget for the protocol"
params: 
  num_rna_samples: 4
  num_dna_samples: 160
  num_pools: 1
  num_seq_lib: 1
---

# Before beginning this protocol, determine your desired sequencing fragment length.

MRS creates a budget by going through each step of the protocol and creating a supply list.  

These lists are read in here and the total numbers of supplies are summed.

These sums are compared against a list read in from the google doc "inventory".  This doc contains item names that match the names used in the supply list for easy merging.

```{r setup}
library(tidyverse)
library(rmarkdown)
library(here)
library(googledrive)
library(googlesheets4)
```

# EecSeq Lab Protocol

The Expressed Exome Capture Sequencing protocol is designed to create exome capture probes directly from RNA.  The probes are then used from hybrid capture of exome DNA sequences, allowing for genotyping of alleles at expressed genes.

#### This protocol is still in the process of being optimized.  Please contact Jon Puritz (jpuritz@uri.edu) for more information before starting!

Before you begin this protocol, you should order the custom oligos required for mRNA library prep and genomic DNA library prep and anneal them. See [Anneal RNA adapters](#anneal-rna-adapters) and [Anneal DNA Adapters](#anneal-adapters)


## RNA Prep
```{r rna-extr, include=FALSE}
render(here("protocols", "rna_extraction_tri.Rmd"), params = list(
  num_samples = params$num_rna_samples
))
```

	
### Quantify all RNA samples
```{r qubit2, include=FALSE}
render(here("protocols", "Qubit-HS.Rmd"), params = list(
  num_samples = params$num_rna_samples
))
qubit2 <- qubit
rm(qubit)
```


## Visualize RNA on BioAnalyzer

```{r bioanalyzer2, include=FALSE}
render(here("protocols", "bioanalyzer.Rmd"), params = list(
  num_samples = params$num_rna_samples))
bioanalyzer2 <- bioanalyzer
rm(bioanalyzer)
```


## Stranded mRNA-Seq Library Prep
### Using KAPA Stranded mRNA-Seq Kit using 1/2 rxn volumes

### Anneal RNA Adapters
```{r anneal, include=FALSE}
render(here("protocols", "anneal.Rmd"), params = list(
  initial_conc_uM=100), run_pandoc = F)
anneal <- anneal %>% 
  # make for 4 adapters
  mutate(quantity = quantity * 4)

# make annealing stock for all 4 adapters
render(here("protocols", "annealing_stock.Rmd"), params = list(
  final_vol_ul = 15000), run_pandoc = F)

# make working stock for all 4 adapters
render(here("protocols", "adapter_working_stock.Rmd"), params = list(
  working_conc_nM = 140), run_pandoc = F)
working <- working %>% 
  # make for 4 adapters
  mutate(quantity = quantity * 4)


```

### mRNA Capture
```{r bead-wash, include=FALSE}
render(here("protocols", "mrna_capture.Rmd"), params = list(
  num_samples = params$num_rna_samples), run_pandoc = F)
```

### Synthesis
```{r synthesis, include = FALSE}
render(here("protocols", "synthesis_eecseq.Rmd"), params = list(
  num_samples = params$num_rna_samples), run_pandoc = F)
```

### Cleanup  - instead of eluting in water or TE, elute in 15 ul A-Tailing buffer
```{r ampure3, include=FALSE}
render(here("protocols", "ampure.Rmd"), params = list(
  num_samples = params$num_rna_samples, 
  sample_vol_ul = 30,
  ampure_vol_ul = 54), run_pandoc = F)
ampure3 <- ampure
rm(ampure)
```

### SAFE STOPPING POINT

#### A-Tailing
```{r a-tailing, include=FALSE}
render(here("protocols", "a-tailing.Rmd"), params = list(
  num_samples = params$num_rna_samples,
  stopped = "yes"), run_pandoc = F)
```

### RNA Adapter Ligation

```{r rna-ligation, include=FALSE}
render(here("protocols", "rna_ligation_eecseq.Rmd"), params = list(num_samples = params$num_rna_samples), run_pandoc = F)
```


### Post-Ligation Cleanup
```{r ampure4, include=FALSE}
render(here("protocols", "ampure.Rmd"), params = list(
  num_samples = params$num_rna_samples, 
  sample_vol_ul = 35,
  ampure_vol_ul = 35), run_pandoc = F)
ampure4 <- ampure
rm(ampure)
```

### Safe Stopping Point
The solution with resuspended beads can be stored at 4 °C for up to 24 hours. Do not freeze the beads, as this can result in dramatic loss of DNA. When ready, proceed to --2nd Post-Ligation Cleanup--.

---

### 2nd Post-Ligation Cleanup
```{r ampure5, include=FALSE}
render(here("protocols", "ampure.Rmd"), params = list(
  num_samples = params$num_rna_samples, 
  sample_vol_ul = 25,
  ampure_vol_ul = 25), run_pandoc = F)
ampure5 <- ampure
rm(ampure)
```


### SAFE STOPPING POINT
The purified, adapter-ligated library DNA may be stored at 4 °C for up to 1 week, or frozen at -20 °C for up to 1 month. When ready, proceed to --Library Amplification--.

---
### Library Amplificiation

```{r rna-amp, include=FALSE}
render(here("protocols", "rna_amp_eecseq.Rmd"), params = list(
  num_samples = params$num_rna_samples
), run_pandoc = F)
```


### Library Amplification Cleanup
```{r ampure6, include=FALSE}
render(here("protocols", "ampure.Rmd"), params = list(
  num_samples = params$num_rna_samples, 
  sample_vol_ul = 25,
  ampure_vol_ul = 25), run_pandoc = F)
ampure6 <- ampure
rm(ampure)
```

## Quant libraries
```{r qubit3, include=FALSE}
render(here("protocols", "Qubit-HS.Rmd"), params = list(
  num_samples = params$num_rna_samples
), run_pandoc = F)
qubit3 <- qubit
rm(qubit)
```

---

### Safe Stopping Point
This is a safe stopping point. If you are stopping, store your sample at ‐15° to ‐25°C.

---


### DSN Normalization

DSN normalization is critical ensuring an even distribution of coverage across probes.  There are a genes that are highly expressed in all cells and DSN normalization helps to remove these high abundance probes and transcripts.  

#### DSN needs to be properly dilued and should be tested for activity levels before proceeding

#### The protocol below was taken from Illumina's recommendations [LINK](https://support.illumina.com/content/dam/illumina-support/documents/myillumina/7836bd3e-3358-4834-b2f7-80f80acb4e3f/dsn_normalization_sampleprep_application_note_15014673_c.pdf)
#### Reagents

| Reagent| Supplier|
|----------|--------------|
|1 M HEPES buffer solution|Invitrogen, part # 15630‐080 |
|5 M NaCl solution|Ambion, part # AM9760G|
|KAPA HiFi HotStart PCR kit with dNTPs|Kapa, part #KK2502|
|Strip tubes|General lab supplier|
|DSN Kit|Evrogen, part # EA001 Sigma Aldrich, part # E7023|
|Ethanol 200 proof (absolute) for molecular biology (500 ml)|AB, part # 4333764F|
|PCR Primer PE 1.0|Included in Kapa stranded mRNA kit|
|PCR Primer PE 2.0|Included in Kapa stranded mRNA kit|
|SPRI beads|Agencourt AMPure, part # 29152; KAPA Pure Beads, part #KK8000|
|Nuclease-free water|General lab supplier|

#### Equipment
- Thermocycler
- Magentic stand compatible with strip tubes

#### Procedure

- First pool individual RNA libraries in equal quantities to create a single pool of 500 ng.
	- For example pool 125 ng each of four individual libraries.

- Create a 4X hybridization solution

|Component|Volume|
|---------|------|
|1 M HEPES buffer solution| 200 𝜇l|
|5 M NaCl solution| 400 𝜇l|
|Nuclease‐free water| 400 𝜇L|
|--Total Volume--|--1000 𝜇--L|

- Use two thermocyclers and set one to hold at 68°C
- Prepare the following reaction mix in a separate, sterile, nuclease‐free 200 ul PCR tube on ice for each sample to be normalized.

|Component|Volume|
|---------|------|
|Sample library (500 ng)| 13.5 𝜇l|
|4X Hybridization buffer| 4.5 𝜇l|
|--Total Volume Per Sample--|--18 𝜇--L|

- Gently pipette the entire volume up and down 10 times, then centrifuge briefly to mix.
- Transfer the entire volume of reaction mix directly to the bottom of a new, sterile, nuclease‐free 200 ul PCR tube, using a pipette. Do not let the sample contact the side of the tube during the process.
- Incubate the reaction mix tube on the thermal cycler using the following PCR cycling conditions:

|Step|Temp|Duration|
|----|----|--------|
|Initial denaturation|98 °C|2 min|
|Treatment|68 °C|5 hours|

- --Caution--- Following incubation, keep the thermal cycler lid closed and the temperature held at 68°C. Do not remove the reaction mix tube from thermal cycler prior to and during DSN treatment.
- Dilute the 10X DSN Master buffer supplied in the DSN kit to 2X with nuclease‐ free water
- Pre‐heat the 2X DSN buffer on the pre‐heated heat block at 68°C.
  - --Note:-- Do not remove the 2X DSN buffer from the heat block during DSN treatment.
- Quickly add 20 ul of pre‐heated 2X DSN buffer to the first reaction mix tube.
- With the reaction mix tube remaining within the thermal cycler, gently pipette the entire volume up and down 10 times to mix thoroughly using a pipette set to 40 ul.
  - --Note--:Pipette the solution directly to the bottom of the PCR tube and do not let the sample contact the side of the tube during the process.
  - --Note--: It is important to keep the thermal cycler closed, except for the time necessary to add the 2X DSN buffer and mix. When preparing more than one reaction mix tube, keep the thermal cycler lid closed when extracting the 2X DSN buffer from its tube, then open the thermal cycler lid only for the time necessary to add and mix the 2X DSN buffer.
- Repeat steps 2 and 3 for each reaction mix tube.
- Incubate the reaction mix tubes on the thermal cycler at 68°C for 10 minutes.
- Quickly add 2 ul of DSN enzyme to the first reaction mix tube using a 2 ul pipette.
- With the reaction mix tube remaining within the thermal cycler, gently pipette the entire volume up and down 10 times to mix thoroughly using a pipette set to 40 ul.
  - --Note--:Pipette the solution directly to the bottom of the PCR tube and do not let the sample contact the side of the tube during the process.
- Repeat steps 6 and 7 for each reaction mix tube.
- Incubate the reaction mix tubes on the thermal cycler at 68°C for 25 minutes.
- Add 40 ul of 2X DSN stop solution to each reaction mix tube. Gently pipette the entire volume up and down to mix thoroughly, then place the tubes on ice.

---

### Safe Stopping Point
This is a safe stopping point. If you are stopping, store your sample at ‐15° to ‐25°C.

---
### SPRI Cleanup

- Perform a 1X SPRI® cleanup by combining the following:

|Component|Volume|
|---------|------|
|DSN Treated Library| 80 ul|
|Agencourt® AMPure® XP reagent| 160 ul|
|--Total Volume--| --240 ul--|

- Thoroughly resuspend the beads by pipetting up and down multiple times.
- Incubate the plate/tube at room temperature for 5 – 15 min to allow the DNA to bind to the beads.
- Place the plate/tube on a magnet to capture the beads. Incubate until the liquid is clear.
- Carefully remove and discard 235 ul of supernatant.
- Keeping the plate/tube on the magnet, add 200 ul of 80% ethanol.
- Incubate the plate/tube at room temperature for ≥30 sec.
- Carefully remove and discard the ethanol.
- Keeping the plate/tube on the magnet, add 200 ul of 80% ethanol.
- Incubate the plate/tube at room temperature for ≥30 sec.
- Carefully remove and discard the ethanol. Try to remove all residual ethanol without disturbing the beads.
- Dry the beads at room temperature, until all of the ethanol has evaporated. --Caution: over-drying the beads may result in dramatic yield loss.--
- Remove the plate/tube from the magnet.
- Thoroughly resuspend the beads in 25 ul of 10 mM Tris-HCl (pH 8.0).
- Incubate the plate/tube at room temperature for 2 min to allow the DNA to elute off the beads.
- Place the plate/tube on a magnet to capture the beads. Incubate until the liquid is clear.
- Transfer 24 ul of the clear supernatant to a new plate/tube and proceed to next step.

### PCR Enrichment of DSN Normalized Library

|Component|Volume|
|---------|------|
|DSN Treated Library | 30 ul|
|2X KAPA HiFi HotStart ReadyMix| 25 ul|
|10X KAPA Library Amplification Primer Mix| 5 ul|
|--Total Volume per sample--| --50 ul--|

- Mix well by pipetting up and down several times
- Amplify the library using the following thermal cycling protocol:

|Step|Temp|Duration|Cycles|
|----|----|--------|------|
|Initial denaturation|98 °C|45 sec|1|
|Denaturation|98 °C|15 sec|12|
|Annealing-|60 °C|30 sec|12|
|Extension|72 °C|30 sec|12|
|Final Extension|72 °C|5 min|1|
|Hold|10 °C | ∞|1|

### SPRI Cleanup of DSN Normalized Library

- Perform a 1.6X SPRI® cleanup by combining the following:

|Component|Volume|
|---------|------|
|Enriched DSN Library| 50 ul|
|Agencourt® AMPure® XP reagent| 80 ul|
|--Total Volume--| --130 ul--|

- Thoroughly resuspend the beads by pipetting up and down multiple times.
- Incubate the plate/tube at room temperature for 5 – 15 min to allow the DNA to bind to the beads.
- Place the plate/tube on a magnet to capture the beads. Incubate until the liquid is clear.
- Carefully remove and discard 115 ul of supernatant.
- Keeping the plate/tube on the magnet, add 200 ul of 80% ethanol.
- Incubate the plate/tube at room temperature for ≥30 sec.
- Carefully remove and discard the ethanol.
- Keeping the plate/tube on the magnet, add 200 ul of 80% ethanol.
- Incubate the plate/tube at room temperature for ≥30 sec.
- Carefully remove and discard the ethanol. Try to remove all residual ethanol without disturbing the beads.
- Dry the beads at room temperature, until all of the ethanol has evaporated. --Caution: over-drying the beads may result in dramatic yield loss.--
- Remove the plate/tube from the magnet.
- Thoroughly resuspend the beads in 22 ul of 10 mM Tris-HCl (pH 8.0).
- Incubate the plate/tube at room temperature for 2 min to allow the DNA to elute off the beads.
- Place the plate/tube on a magnet to capture the beads. Incubate until the liquid is clear.
- Transfer 22 ul of the clear supernatant to a new plate/tube and proceed to next step.

## Quant libraries
--Procedure (Standard HS DNA protocol)--
- Set up the required number of 0.5-mL tubes for standards and samples. The Qubit® RNA HS Assay requires 2 standards.
- Label the tube lids.
- Prepare the Qubit® working solution by diluting the Qubit® DNA HS Reagent 1:200 in Qubit® DNA HS Buffer. Use a clean plastic tube each time you prepare Qubit® working solution. --Do not mix the working solution in a glass container.--
- Add 190 uL of Qubit® working solution to each of the tubes used for standards.
- Add 10 uL of each Qubit® standard to the appropriate tube, then mix by vortexing 2–3 seconds. Be careful not to create bubbles.
- Add Qubit® working solution to individual assay tubes so that the final volume in each tube after adding sample is 200 uL.
- Add each sample to the assay tubes containing the correct volume of Qubit® working solution, then mix by vortexing 2–3 seconds. The final volume in each tube should be 200 uL.
- Allow all tubes to incubate at room temperature for 2 minutes.
- On the Home screen of the Qubit® 3.0 Fluorometer, press DNA, then select DNA: High Sensitivity as the assay type. The “Read standards” screen is displayed. Press Read Standards to proceed.
- Insert the tube containing Standard #1 into the sample chamber, close the lid, then press Read standard. When the reading is complete (~3 seconds), remove Standard #1.
- Insert the tube containing Standard #2 into the sample chamber, close the lid, then press Read standard. When the reading is complete, remove Standard #2.
- Press Run samples.
- On the assay screen, select the sample volume and units
- Insert a sample tube into the sample chamber, close the lid, then press Read tube. When the reading is complete (~3 seconds), remove the sample tube.
- Repeat step last step until all samples have been read

---

## Split finished cDNA library for each sample into two vials
- One tube for sequencing
- One tube for probe synthesis

#### Note that this step is optional as sequening the probes directly is not necessary for exome capture analysis.  Alternatively, a portion of the mRNA library can be saved before DSN normalization for sequencing.  

---

### Safe Stopping Point
This is a safe stopping point. If you are stopping, store your sample at ‐15° to ‐25°C.

---

## Probe Synthesis

Review quantifications for probes.  Ideally, there should be about 500 ng of probes per capture.  If not enough of probes is obtained, the PCR product can be re-amplified.

### Remove adapters from cDNA

#### Materials needed
| Reagent                                | Producer         | Catalog #      |            
|----------------------------------------|------------------|-----------|
|Mung Bean Nuclease| NEB| M0250S|
|SalI-HF| NEB| R3138T|
|Agencourt AMPure XP  |Beckman Coulter   | A63881|

### Procedure
- Setup a restriction digest using 1 ug of DSN library

|Component|Volume|
|---------|------|
| 10X Cutsmart Buffer| 4 ul|
|SalI-HF Enzyme (100 units)| 1 ul |
| Molecular Grade H20| 22.75 ul|
| DSN Enriched Library| 12.25 ul|
|--Total Volume--| --40 ul--|

- Incubate reactions in thermocycler at 37°C for at least 8 hours, prefereably 12-16 hours.

- --Using the same tubes from the previous step-- combine the following:

|Component|Volume|
|---------|------|
|Digested Library| 40 ul|
|10X Mung Bean Nuclease buffer| 4.5 ul|
|Mung Bean Nuclease (10 units per ul)| 0.5 ul|
|--Total Volume--| --45 ul--|

- Incubate at 30°C for 30 minutes

- Perform a 1.8X SPRI cleanup by combining the following:

#### This step may be possbile to skip and proceed directly to the 1.5X SPRI Cleanup.

### It is important to check for the digested adapters.  See trace below with adapter peaks:

![alt text](/ProbesandDigested.png)

#### If your trace looks like this, perform the additional 1.5X SPRI Cleanup below

|Component|Volume|
|---------|------|
|MBN reaction| 45 ul|
|Agencourt® AMPure® XP reagent| 81 ul|
|--Total Volume--| --126 ul--|

- Thoroughly resuspend the beads by pipetting up and down multiple times.
- Incubate the plate/tube at room temperature for 5 – 15 min to allow the DNA to bind to the beads.
- Place the plate/tube on a magnet to capture the beads. Incubate until the liquid is clear.
- Carefully remove and discard 115 ul of supernatant.
- Keeping the plate/tube on the magnet, add 200 ul of 80% ethanol.
- Incubate the plate/tube at room temperature for ≥30 sec.
- Carefully remove and discard the ethanol.
- Keeping the plate/tube on the magnet, add 200 ul of 80% ethanol.
- Incubate the plate/tube at room temperature for ≥30 sec.
- Carefully remove and discard the ethanol. Try to remove all residual ethanol without disturbing the beads.
- Dry the beads at room temperature, until all of the ethanol has evaporated. --Caution: over-drying the beads may result in dramatic yield loss.--
- Remove the plate/tube from the magnet.
- Thoroughly resuspend the beads in 22 ul of 10 mM Tris-HCl (pH 8.0). Volume needed depends on the number of captures. Calculate 10 ul per capture plus an aliquot for checking the probes concentration using Qubit.
- Incubate the plate/tube at room temperature for 2 min to allow the DNA to elute off the beads.
- Place the plate/tube on a magnet to capture the beads. Incubate until the liquid is clear.
- Transfer 20 ul of the clear supernatant to a new plate/tube and proceed to next step.

---

### Safe Stopping Point
This is a safe stopping point. If you are stopping, store your sample at ‐15° to ‐25°C.

---

- Perform a 1.5X SPRI cleanup by combining the following:

|Component|Volume|
|---------|------|
|MBN reaction| 22 ul|
|Agencourt® AMPure® XP reagent|33 ul|
|--Total Volume--| --55 ul--|

- Thoroughly resuspend the beads by pipetting up and down multiple times.
- Incubate the plate/tube at room temperature for 5 – 15 min to allow the DNA to bind to the beads.
- Place the plate/tube on a magnet to capture the beads. Incubate until the liquid is clear.
- Carefully remove and discard 115 ul of supernatant.
- Keeping the plate/tube on the magnet, add 200 ul of 80% ethanol.
- Incubate the plate/tube at room temperature for ≥30 sec.
- Carefully remove and discard the ethanol.
- Keeping the plate/tube on the magnet, add 200 ul of 80% ethanol.
- Incubate the plate/tube at room temperature for ≥30 sec.
- Carefully remove and discard the ethanol. Try to remove all residual ethanol without disturbing the beads.
- Dry the beads at room temperature, until all of the ethanol has evaporated. --Caution: over-drying the beads may result in dramatic yield loss.--
- Remove the plate/tube from the magnet.
- Thoroughly resuspend the beads in 31 ul of 10 mM Tris-HCl (pH 8.0). Volume needed depends on the number of captures. Calculate 10 ul per capture plus an aliquot for checking the probes concentration using Qubit.
- Incubate the plate/tube at room temperature for 2 min to allow the DNA to elute off the beads.
- Place the plate/tube on a magnet to capture the beads. Incubate until the liquid is clear.
- Transfer 30 ul of the clear supernatant to a new plate/tube and proceed to next step.

---

### Safe Stopping Point
This is a safe stopping point. If you are stopping, store your sample at ‐15° to ‐25°C.

---

## Biotin Labeling

### Materials needed

| Reagent                                | Producer         | Catalog #      |            
|----------------------------------------|------------------|-----------|
|DecaLabel™ Biotin DNA Labeling Kit |Thermo Scientific | FERK0651   |                     

### Procedure

- Add the following components into 1.5 ml microcentrifuge tube:

|Component|Volume|
|---------|------|
|RE and MBN treated DSN Library| 20 ul|
|Decanucleotide in 5X Reaction Buffer| 10 ul|
| Water, nuclease-free| 14 ul|
|--Total Volume--| --44 ul--|

- Vortex the tube and spin down in a microcentrifuge for 3-5 s
- Incubate the tube in a boiling water bath for 5-10 min and cool it on ice. Spin down quickly.
- Add the following components in the same tube:

|Component|Volume|
|---------|------|
|Biotin Labeling Mix| 5 ul|
|Klenow fragment, exo– (5 u)| 1 ul|
|--Total Volume--| --50 ul--|

- Shake the tube and spin down in a microcentrifuge for 3-5 s. 
- Incubate for 12-20 hours at 37°C. 

---
#### Optional:Control reaction
- Add the following components into 1.5 ml microcentrifuge tube:

|Component|Volume|
|---------|------|
|Control Template, 10 ng/ul | 25 ul|
|Decanucleotide in 5X Reaction Buffer| 10 ul|
| Water, nuclease-free| 9 ul|
|--Total Volume--| --44 ul--|

- Vortex the tube and spin down in a microcentrifuge for 3-5 s
- Incubate the tube in a boiling water bath for 5-10 min and cool it on ice. Spin down quickly.
- Add the following components in the same tube:

|Component|Volume|
|---------|------|
|Biotin Labeling Mix| 5 ul|
|Klenow fragment, exo– (5 u)| 1 ul|
|--Total Volume--| --50 ul--|

---

- Perform a 1.5X SPRI® cleanup by combining the following:

|Component|Volume|
|---------|------|
|Biotin reaction| 50 ul|
|Agencourt® AMPure® XP reagent| 75 ul|
|--Total Volume--| --125 ul--|

- Thoroughly resuspend the beads by pipetting up and down multiple times.
- Incubate the plate/tube at room temperature for 5 – 15 min to allow the DNA to bind to the beads.
- Place the plate/tube on a magnet to capture the beads. Incubate until the liquid is clear.
- Carefully remove and discard 115 ul of supernatant.
- Keeping the plate/tube on the magnet, add 200 ul of 80% ethanol.
- Incubate the plate/tube at room temperature for ≥30 sec.
- Carefully remove and discard the ethanol.
- Keeping the plate/tube on the magnet, add 200 ul of 80% ethanol.
- Incubate the plate/tube at room temperature for ≥30 sec.
- Carefully remove and discard the ethanol. Try to remove all residual ethanol without disturbing the beads.
- Dry the beads at room temperature, until all of the ethanol has evaporated. --Caution: over-drying the beads may result in dramatic yield loss.--
- Remove the plate/tube from the magnet.
- Thoroughly resuspend the beads in 20 ul of 10 mM Tris-HCl (pH 8.0). Volume needed depends on the number of captures. Calculate 10 ul per capture plus an aliquot for checking the probes concentration using Qubit.
- Incubate the plate/tube at room temperature for 2 min to allow the DNA to elute off the beads.
- Place the plate/tube on a magnet to capture the beads. Incubate until the liquid is clear.
- Transfer 19 ul of the clear supernatant to a new plate/tube and proceed to next step.

## Quant Probes
--Procedure (Standard HS DNA protocol)--
- Set up the required number of 0.5-mL tubes for standards and samples. The Qubit® RNA HS Assay requires 2 standards.
- Label the tube lids.
- Prepare the Qubit® working solution by diluting the Qubit® DNA HS Reagent 1:200 in Qubit® DNA HS Buffer. Use a clean plastic tube each time you prepare Qubit® working solution. --Do not mix the working solution in a glass container.--
- Add 190 uL of Qubit® working solution to each of the tubes used for standards.
- Add 10 uL of each Qubit® standard to the appropriate tube, then mix by vortexing 2–3 seconds. Be careful not to create bubbles.
- Add Qubit® working solution to individual assay tubes so that the final volume in each tube after adding sample is 200 uL.
- Add each sample to the assay tubes containing the correct volume of Qubit® working solution, then mix by vortexing 2–3 seconds. The final volume in each tube should be 200 uL.
- Allow all tubes to incubate at room temperature for 2 minutes.
- On the Home screen of the Qubit® 3.0 Fluorometer, press DNA, then select DNA: High Sensitivity as the assay type. The “Read standards” screen is displayed. Press Read Standards to proceed.
- Insert the tube containing Standard #1 into the sample chamber, close the lid, then press Read standard. When the reading is complete (~3 seconds), remove Standard #1.
- Insert the tube containing Standard #2 into the sample chamber, close the lid, then press Read standard. When the reading is complete, remove Standard #2.
- Press Run samples.
- On the assay screen, select the sample volume and units
- Insert a sample tube into the sample chamber, close the lid, then press Read tube. When the reading is complete (~3 seconds), remove the sample tube.
- Repeat step last step until all samples have been read

## Visualize Probes

- Run probes on BioAnalyzer/Tape Station/Fragment analyzer 

They should look like this:

![alt text](/FinalProbes.png)

---

### Safe Stopping Point
This is a safe stopping point. If you are stopping, store your sample at ‐15° to ‐25°C.

---

## Preparation of whole genome libraries 

### Using KAPA HyperPlus Kit with 1/2 reactions

Refer to [manual](https://www.kapabiosystems.com/document/kapa-hyperplus-library-preparation-kit-tds/?dl=1) during procedure (steps below are for notes and comments). 

#### For Puritz and Lotterhos 2017, genomic DNA was sheared to a modal peak of 150 bp using a Covaris sonicator.  The protocol below assumes you are starting with sheared DNA

### Materials Needed

Adapter Oligos:

|Name| 5' to 3' Sequence|
|----------|---------------------------------------------|
|DNA_P1.1.1|	ACACTCTTTCCCTACACGACGCTCTTCCGATCTGCATGG-T|
|DNA_P1.1.2|	ACACTCTTTCCCTACACGACGCTCTTCCGATCTAACCAG-T|
|DNA_P1.1.3|	ACACTCTTTCCCTACACGACGCTCTTCCGATCTCGATCG-T|
|DNA_P1.1.4|	ACACTCTTTCCCTACACGACGCTCTTCCGATCTTCGATG-T|
|DNA_P1.1.5|	ACACTCTTTCCCTACACGACGCTCTTCCGATCTTGCATG-T|
|DNA_P1.1.6|	ACACTCTTTCCCTACACGACGCTCTTCCGATCTCAACCG-T|
|DNA_P1.1.7|	ACACTCTTTCCCTACACGACGCTCTTCCGATCTGGTTGG-T|
|DNA_P1.1.8|	ACACTCTTTCCCTACACGACGCTCTTCCGATCTAAGGAG-T|
|DNA_P1.2.1|	PC-CATGCAGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT|
|DNA_P1.2.2|	PC-TGGTTAGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT|
|DNA_P1.2.3|	PC-GATCGAGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT|
|DNA_P1.2.4|	PC-ATCGAAGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT|
|DNA_P1.2.5|	PC-ATGCAAGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT|
|DNA_P1.2.6|	PC-GGTTGAGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT|
|DNA_P1.2.7|	PC-CAACCAGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT|
|DNA_P1.2.8|	PC-TCCTTAGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT|
|DNA_P2.1|	P-GATCGGAAGAGCGAGAACAA|
|DNA_P2.2|	GTGACTGGAGTTCACACGTGTGCTCTTCCGATC-T|

PCR Primers:

|Name| 5' to 3' Sequence|
|----------|---------------------------------------------|
|PCR1_P5_primer|AATGATACGGCGACCACCGAGATCTACACTCTTTCCCTACACGACGCTCTTCCGATC-T|
|PCR2_01_ATCACG|CAAGCAGAAGACGGCATACGAGATCGTGATGTGACTGGAGTTCAGACGTGTG-C|
|PCR2_02_CGATGT|CAAGCAGAAGACGGCATACGAGATACATCGGTGACTGGAGTTCAGACGTGTG-C|
|PCR2_03_TTAGGC|CAAGCAGAAGACGGCATACGAGATGCCTAAGTGACTGGAGTTCAGACGTGTG-C|
|PCR2_04_TGACCA|CAAGCAGAAGACGGCATACGAGATTGGTCAGTGACTGGAGTTCAGACGTGTG-C|
|PCR2_05_ACAGTG|CAAGCAGAAGACGGCATACGAGATCACTGTGTGACTGGAGTTCAGACGTGTG-C|
|PCR2_06_GCCAAT|CAAGCAGAAGACGGCATACGAGATATTGGCGTGACTGGAGTTCAGACGTGTG-C|
|PCR2_07_CAGATC|CAAGCAGAAGACGGCATACGAGATGATCTGGTGACTGGAGTTCAGACGTGTG-C|
|PCR2_08_ACTTGA|CAAGCAGAAGACGGCATACGAGATTCAAGTGTGACTGGAGTTCAGACGTGTG-C|
|PCR2_09_GATCAG|CAAGCAGAAGACGGCATACGAGATCTGATCGTGACTGGAGTTCAGACGTGTG-C|
|PCR2_10_TAGCTT|CAAGCAGAAGACGGCATACGAGATAAGCTAGTGACTGGAGTTCAGACGTGTG-C|
|PCR2_11_GGCTAC|CAAGCAGAAGACGGCATACGAGATGTAGCCGTGACTGGAGTTCAGACGTGTG-C|
|PCR2_12_CTTGTA|CAAGCAGAAGACGGCATACGAGATTACAAGGTGACTGGAGTTCAGACGTGTG-C|

### Anneal Adapters
Single-stranded oligos need to be annealed with their appropriate partner before ligation. 
 
- To create Adapter P1, combine each oligo 1.1 with its complementary oligo 1.2 in a 1:1 ratio in working strength annealing buffer (final buffer concentration 1x) for a total annealed adapter concentration of 40uM (for example, if purchased oligos are resuspended to an initial concentration of 100uM, use 40ul oligo 1.1, 40ul oligo 1.2, 10ul 10x annealing buffer and 10ul nuclease-free water). Do the same for oligos 2.1 and 2.2 to create the common adapter P2. 
- In a thermocyler, incubate at 97.5°C for 2.5 minutes, and then cool at a rate of not greater than 3°C per minute until the solution reaches a temperature of 21°C. Hold at 4°C.
- Prepare final working strength concentrations of annealed adapters from this annealed stock. For convenience, it is possible to store the adapters at 4°C while in active use.  --For Puritz and Lotterhos 2017, a working stock of 40 uM was used, leading to a final adapter:insert molar ratio of ~ 50:1.--

### Note about multiplexing and DNA adapters

The above adapters have inline barcodes which will appear in the sequences and need to be demultiplexed bioinformatically.

|Adapter| Inline Barcode|
|-------|---------------|
|1|GCATG|
|2|AACCA|
|3|CGATC|
|4|TCGAT|
|5|TGCAT|
|6|CAACC|
|7|GGTTG|
|8|AAGGA|

These inline barcodes can be combined with the 12 PCR primers that add a P7 Illumina index which is read directly in a sepeartate sequencing run and is returned to the user in index-labeled files.  With the above design 96 different individuals or pools could be multiplexed on a single lane.

Alternatively, any TruSeq style DNA adapters could be used with this protocol as long as they match with the [Blockling Oligos](#blocking-oligos) used during [Probe Hybridization](#hybridization)
---

### Procedure

### End repair
- Adjust sample volume of 500 ng fragmented DNA to 22.5 ul.
- Add the following to each sample:

|Component|Volume|
|---------|------|
|KAPA Frag Buffer (10X) | 2.5|
|End Repair & A-Tailing Buffer- | 3.5 ul|
|End Repair & A-Tailing Enzyme Mix- | 1.5 ul|
|Fragmented, double-stranded DNA| 25 ul |
|--Total Volume--| --30 ul--|

  - The buffer and enzyme mix should preferably be pre-mixed and added in a single pipetting step. 
    - Premixes are stable for ≤24 hrs at room temperature, for ≤3 days at 4°C, and for ≤4 weeks at -20°C
  - Vortex gently and spin down briefly. Return the reaction plate/tube(s) to ice. Proceed immediately to the next step.
  - Incubate in a thermocycler programmed as outlined below. A heated lid is required for this step. If possible, set the temperature of the heated lid to ~85°C (instead of the usual 105°C).

|Step|Temp|Time|
|----|----|--------|
|End repair and A-tailing 1|20 °C|30 min|
|End repair and A-tailing 2|65 °C|30 min|
|Hold|10 °C | ∞|

- Notes
  - A heated lid is required for this incubation. If possible, set the temperature of the lid at 85°C, instead of the usual ~105°C.
  - If proceeding to the adapter ligation reaction setup without any delay, the reaction may be cooled to 20°C instead of 4°C.

### Adapter ligation
- Dilute adapter stocks to the appropriate concentration, as outlined below:

|Fragmented DNA| Adapter stock concentration | Adapter:insert molar ratio|
|--------------|-----------------------------|---------------------------|
| 1 ug|15 uM|10:1|
|500 ng|15 uM|20:1|
|250 ng|15 uM|40:1|
|100 ng|15 uM|100:1|
|50 ng|15 uM|200:1|
|25 ng|15 uM|200:1|
|10 ng|15 uM|200:1|
|5 ng|15 uM|200:1|
|2.5 ng|15 uM|200:1|
|1 ng|15 uM|200:1|

--For Puritz and Lotterhos (2017), a working stock of 40 uM was used, leading to a final adapter:insert molar ratio of ~ 50:1.--

- In the same plate/tube(s) in which end repair and A-tailing was performed, assemble each adapter ligation reaction as follows:

|Component|Volume|
|---------|------|
|End repair and A-tailing reaction product| 30 ul|
|P1 Adapter stock (concentration as required) --Barcode containing oligo-- | 1.25 ul|
|P2 Adapter stock (concentration as required) | 1.25 ul|
|PCR-grade water-| 2.5 ul |
|Ligation Buffer-| 15 ul |
|DNA ligase-| 5 ul |
|--Total Volume--| --55 ul--|

- Notes
  - --Each P1 adapter has a unique barcode-- these barcodes can be combined with Illumina indices (added later via PCR) for high levels of multiplexing. 
  - The water, buffer and ligase enzyme should preferably be premixed and added in a single pipetting step. Premixes are stable for ≤24 hrs at room temperature, for ≤3 days at 4°C, and for ≤4 weeks at -20°C.
  
- Mix thoroughly and centrifuge briefly.
- Incubate at 20°C for 60 min.
  - Note: to achieve higher conversion rates and library yields, particularly for low-input samples, consider increasing the ligation time—to a maximum of 4 hrs at 20°C, or overnight at 4°C. Please note that longer ligation times may lead to increased levels of adapter-dimer. Adapter concentrations may have to be optimized if ligation times are extended signi cantly.
- Proceed immediately to the next step.

### Post-ligation Cleanup
```{r kapa-clean, include=FALSE}
 render(here("protocols", "kapa-clean.Rmd"), params = list(
  sample_vol_ul = 55,
  bead_vol_ul = 44, 
  num_samples = params$num_dna_samples), run_pandoc = F)
```

### Quant samples
```{r quant, include=FALSE}
if(params$num_dna_samples >= 20){
  render(here("protocols", "plate-reader.Rmd"), params = list(
  num_samples = params$num_dna_samples), run_pandoc = F)
}else{
  render(here("protocols", "Qubit-HS.Rmd"), params = list(
  num_samples = params$num_dna_samples), run_pandoc = F)
  qubit2 <- qubit
  rm(qubit)
}
```



### Pool samples to be used with in the same index/capture.  
```{r pool,include=FALSE}
render(here("protocols", "pool.Rmd"), params = list(
  num_samples_in = params$num_dna_samples, 
  num_pools_out = params$num_seq_lib
))
```

Each P1 adapter has a unique barcode sequence. These barcodes can be combined with Illumina indices (added later via PCR) for high levels of multiplexing within in a single lane of Illumina sequencing.  At this point samples with different barcodes but the same Illumina index can be pooled before amplification.  The current version of the protocol supports pooling 12 individuals per Illumina index, along with 12 unique Illumina Indices.

### Library Amplification

```{r amp, include=FALSE}
render(here("protocols", "lib-amp_eecseq.Rmd"), params = list(
  num_seq_lib = params$num_seq_lib))
```


### Post-amplification Cleanup
```{r ampure2, include=FALSE}
render(here("protocols", "ampure.Rmd"), params = list(
  num_samples = params$num_seq_lib, 
  sample_vol_ul = 25,
  ampure_vol_ul = 25))
ampure2 <- ampure
rm(ampure)
```
- Resuspend in 15 ul of 10 mM Tris or water

---

### Safe Stopping Point
This is a safe stopping point. If you are stopping, store your sample at ‐15° to ‐25°C.

---
## Hybridization and Capture

### Hybridization

```{r hybrid, include=FALSE}

render(here("protocols", "hybridization_eecseq.Rmd"), params = list(
  lib_size_ng = 500, 
  num_seq_lib = 1))

# may need to make 500mM EDTA stock solution - 50mL
render(here("protocols", "edta_stock.Rmd"), params = list(
  final_conc_M = 0.500,
  final_vol_l = 0.050))

# may need to make 10% SDS stock solution - 50mL
render(here("protocols", "sds_stock.Rmd"), params = list(
  final_conc_perc = 10,
  final_vol_ml = 50))

# may need to make TEN stock solution - 50 ml
render(here("protocols", "ten_stock.Rmd"), params = list(
 final_vol_ul = 50000))

# may need to make various SSC / 0.1% SDS stock solution - 50 ml
render(here("protocols", "ssc_sds_stock_eecseq.Rmd"), params = list(
 final_vol_ul = 500, 
 perc_ssc = 1))
wash1 <- wash
rm(wash)

render(here("protocols", "ssc_sds_stock_eecseq.Rmd"), params = list(
 final_vol_ul = 500, 
 perc_ssc = 0.5))
wash2 <- wash
rm(wash)

render(here("protocols", "ssc_sds_stock_eecseq.Rmd"), params = list(
 final_vol_ul = 500, 
 perc_ssc = 0.1))
wash3 <- wash
rm(wash)
```
					

### Library re-amplification
```{r reamp, include=FALSE}
render(here("protocols", "lib-reamp_eecseq.Rmd"), params = list(
  num_seq_lib = params$num_seq_lib))
```

### Perform a 1X SPRI® cleanup by combining the following:
```{r ampure, include = FALSE}
render(here("protocols", "ampure.Rmd"), params = list(
  num_samples = params$num_seq_lib, 
  sample_vol_ul = 25,
  ampure_vol_ul = 25))

# may need to make 10mM pH 8 Tris-HCl stock solution - 50mL
render(here("protocols", "tris-hcl_stock.Rmd"), params = list(
  final_conc_M = 0.010,
  pH = 8.0,
  final_vol_l = 0.050,
  mol_wt = 121.1))
```


### Quant samples
```{r qubit, include = FALSE}
render(here("protocols", "Qubit-HS.Rmd"), params = list(
  num_samples = params$num_seq_lib))
```

### Verify  
- Run samples on BioAnalyzer/Tape Station/Fragment analyzer  
- check with Nicole Wagner in Debashish's lab for current pricing.  This assumes you are using a High Sensitivity chip (max 7kb per library).
```{r bioanalyzer, include=FALSE}
render(here("protocols", "bioanalyzer.Rmd"), params = list(
  num_samples = params$num_seq_lib))
```


# Combine the lists
```{r combine}
all_lists <- bind_rows(amp, ampure, ampure2, anneal, annealing, bioanalyzer, clean, dyna, edta, hybrid, kapa_clean, plate, qubit, reamp, sds, TEN, tris_hcl, wash3, wash1, wash2, working)

all_sum <- all_lists %>% 
  group_by(item) %>% 
  summarise(quantity = ceiling(sum(quantity))) %>% 
  # adjust the size of the ampure purchase if needed
  mutate(item = ifelse(item == "ampure_ul", 
                       ifelse(quantity > 60000, "ampure_450", "ampure_60"),item))%>% 
  arrange()
```

# Merge with inventory costs
```{r inventory}
# enter code here to look up inventory on google drive
inventory_file <- drive_get("~/Pinsky Lab Orders/Inventory")
inventory_raw <- read_sheet(inventory_file, sheet = "Inventory", col_types = "c????n???n????")

supplies <- left_join(all_sum, inventory_raw)
```

