---
title: "Create a budget for the protocol"
params: 
  num_rna_samples: 4
  num_dna_samples: 96
  num_pools: 12
  num_seq_lib: 1
  num_rna_lib: 1
  num_dna_adapters: 14
  num_rna_adapters: 4
---

# Before beginning this protocol, determine your desired sequencing fragment length.

MRS creates a budget by going through each step of the protocol and creating a supply list.  

These lists are read in here and the total numbers of supplies are summed.

These sums are compared against a list read in from the google doc "inventory".  This doc contains item names that match the names used in the supply list for easy merging.

```{r setup}
knitr::opts_chunk$set(
	message = FALSE,
	warning = FALSE,
	include = FALSE
)
options(scipen = 999)
library(tidyverse)
library(rmarkdown)
library(here)
library(googledrive)
library(googlesheets4)
```

# EecSeq Lab Protocol

### 1. RNA Extraction
```{r rna-extr}
render(here("protocols", "rna_extraction_tri.Rmd"), params = list(
  num_samples = params$num_rna_samples
))
```

	
### 2. Quantify all RNA samples
```{r quant1}
render(here("protocols", "quant_dna.Rmd"), params = list(
  num_samples = params$num_rna_samples), run_pandoc = F)
  quant1 <- quant
  rm(quant)
```


### 3. Visualize RNA on BioAnalyzer

```{r bioanalyzer1}
render(here("protocols", "bioanalyzer.Rmd"), params = list(
  num_samples = params$num_rna_samples), run_pandoc = F)
bioanalyzer1 <- bioanalyzer %>% 
  # because this run can't share wells with later runs:
  mutate(quantity = quantity + (12-quantity))
rm(bioanalyzer)
```

### 4. Anneal RNA Adapters
```{r anneal1}
render(here("protocols", "anneal.Rmd"), params = list(
  initial_conc_uM=100, 
  p1_adapter_name="sai_universal_stock_adapter_ul",
  p2_adapter_name="sai_indexed_stock_adapter_ul"), run_pandoc = F)
anneal <- anneal %>% 
  # make for 4 adapters
  mutate(quantity = quantity * params$num_rna_adapters, 
         #multiple the indexes by 5 to eke the quantity up enough to register that we need to buy 4 of them.
         quantity = ifelse(grepl("index", item), quantity * 5, quantity))
anneal1 <- anneal
rm(anneal)

# make annealing stock for all 4 adapters
render(here("protocols", "annealing_stock.Rmd"), params = list(
  final_vol_ul = 15000), run_pandoc = F)
```
```{r working1}
# make working stock for all 4 adapters
render(here("protocols", "adapter_working_stock.Rmd"), params = list(
  initial_conc_uM = 40,
  working_conc_uM = 10, 
  final_vol_ul = 100, 
  adpater_stock = "sai_working_adapter_ul"
),run_pandoc = F)
working1 <- working %>% 
  # make for 4 adapters
  mutate(quantity = quantity * params$num_rna_adapters)
rm(working)
```

### 5. mRNA Capture
```{r bead-wash}
render(here("protocols", "mrna-stranded.Rmd"), params = list(
  num_samples = params$num_rna_samples), run_pandoc = F)
```

### 6. Synthesis
```{r synthesis}
render(here("protocols", "synthesis_eecseq.Rmd"), params = list(
  num_samples = params$num_rna_samples), run_pandoc = F)
```

### 7. Cleanup  
- instead of eluting in water or TE, elute in 15 ul A-Tailing buffer
```{r ampure1}
render(here("protocols", "ampure.Rmd"), params = list(
  num_samples = params$num_rna_samples, 
  sample_vol_ul = 30,
  ampure_vol_ul = 54), run_pandoc = F)
ampure1 <- ampure
rm(ampure)
```

### 8. A-Tailing
```{r a-tailing}
render(here("protocols", "a-tailing.Rmd"), params = list(
  num_samples = params$num_rna_samples,
  stopped = "yes"), run_pandoc = F)
```

### 9. RNA Adapter Ligation

```{r rna-ligation}
render(here("protocols", "rna_ligation_eecseq.Rmd"), params = list(num_samples = params$num_rna_samples), run_pandoc = F)
```


### 10 Post-Ligation Cleanup
```{r ampure2}
render(here("protocols", "ampure.Rmd"), params = list(
  num_samples = params$num_rna_samples, 
  sample_vol_ul = 35,
  ampure_vol_ul = 35), run_pandoc = F)
ampure2 <- ampure
rm(ampure)
```

### Safe Stopping Point
The solution with resuspended beads can be stored at 4 °C for up to 24 hours. Do not freeze the beads, as this can result in dramatic loss of DNA. When ready, proceed to --2nd Post-Ligation Cleanup--.

---

### 2nd Post-Ligation Cleanup
```{r ampure3}
render(here("protocols", "ampure.Rmd"), params = list(
  num_samples = params$num_rna_samples, 
  sample_vol_ul = 25,
  ampure_vol_ul = 25), run_pandoc = F)
ampure3 <- ampure
rm(ampure)
```


### SAFE STOPPING POINT
The purified, adapter-ligated library DNA may be stored at 4 °C for up to 1 week, or frozen at -20 °C for up to 1 month. When ready, proceed to --Library Amplification--.

---
### 12. RNA Library Amplificiation

```{r rna-amp}
render(here("protocols", "rna_amp_eecseq.Rmd"), params = list(
  num_samples = params$num_rna_samples
), run_pandoc = F)
```


### 13 Library Amplification Cleanup
```{r ampure4}
render(here("protocols", "ampure.Rmd"), params = list(
  num_samples = params$num_rna_samples, 
  sample_vol_ul = 25,
  ampure_vol_ul = 25), run_pandoc = F)
ampure4 <- ampure
rm(ampure)
```

## 14. Quant libraries
```{r quant2}
render(here("protocols", "quant_dna.Rmd"), params = list(
  num_samples = params$num_rna_samples), run_pandoc = F)
  quant2 <- quant
  rm(quant)
```

---

### Safe Stopping Point
This is a safe stopping point. If you are stopping, store your sample at ‐15° to ‐25°C.

---


### 15. DSN Normalization
```{r dsn}
render(here("protocols", "dsn-norm.Rmd"), params = list(
    num_rna = params$num_rna_samples,
  num_lib = params$num_rna_lib
), run_pandoc = F)

# may need to make 5M NaCl solution
render(here("protocols", "nacl_stock.Rmd"), params = list(
    final_conc_M = 5,
  final_vol_l = 0.050), run_pandoc = F)
```

### 16. Cleanup
```{r ampure5}
render(here("protocols", "ampure.Rmd"), params = list(
  num_samples = params$num_rna_lib, 
  sample_vol_ul = 80,
  ampure_vol_ul = 160), run_pandoc = F)
ampure5 <- ampure
rm(ampure)
```

### 17. DSN PCR
```{r dsn-pcr}
render(here("protocols", "dsn-pcr.Rmd"), params = list(
  num_samples = params$num_rna_lib), run_pandoc = F)
```


### 18. Cleanup
```{r ampure6}
render(here("protocols", "ampure.Rmd"), params = list(
  num_samples = params$num_rna_lib, 
  sample_vol_ul = 50,
  ampure_vol_ul = 80), run_pandoc = F)
ampure6 <- ampure
rm(ampure)
```

### 19. Quantify
```{r quant3}
render(here("protocols", "quant_dna.Rmd"), params = list(
  num_samples = params$num_rna_samples), run_pandoc = F)
  quant3 <- quant
  rm(quant)
```


## Probe Synthesis


### Remove adapters from cDNA
```{r remove-adapters}
render(here("protocols", "remove-adapters_eecseq.Rmd"),params = list(
  num_samples = params$num_rna_lib
), run_pandoc = F)
```


### Bioanalyzer or trace
```{r bioanalyzer2}
render(here("protocols", "bioanalyzer.Rmd"), params = list(
  num_samples = params$num_seq_lib), run_pandoc = F)
bioanalyzer2 <- bioanalyzer  %>% 
  # because this run can't share wells with later runs:
  mutate(quantity = quantity + (12-quantity))
rm(bioanalyzer)
```


### Perform a 1.8X SPRI cleanup by combining the following:
```{r ampure7}
render(here("protocols", "ampure.Rmd"), params = list(
  num_samples = params$num_rna_lib, 
  sample_vol_ul = 45,
  ampure_vol_ul = 81), run_pandoc = F)
ampure7 <- ampure
rm(ampure)
```



### Perform a 1.5X SPRI cleanup by combining the following:
```{r ampure8}
render(here("protocols", "ampure.Rmd"), params = list(
  num_samples = params$num_rna_lib, 
  sample_vol_ul = 22,
  ampure_vol_ul = 33), run_pandoc = F)
ampure8 <- ampure
rm(ampure)
```

### Safe Stopping Point
This is a safe stopping point. If you are stopping, store your sample at ‐15° to ‐25°C.

---

### Biotin Labeling
```{r biotin}
render(here("protocols", "biotin_eecseq.Rmd"),params = list(
  num_samples = params$num_rna_lib
), run_pandoc = F)
```


###  Perform a 1.5X SPRI® cleanup by combining the following:
```{r ampure9}
render(here("protocols", "ampure.Rmd"), params = list(
  num_samples = params$num_rna_lib, 
  sample_vol_ul = 50,
  ampure_vol_ul = 75), run_pandoc = F)
ampure9 <- ampure
rm(ampure)
```


### Quant Probes
```{r quant4}
render(here("protocols", "quant_dna.Rmd"), params = list(
  num_samples = params$num_rna_samples), run_pandoc = F)
  quant4 <- quant
  rm(quant)
```


### Visualize Probes

- Run probes on BioAnalyzer/Tape Station/Fragment analyzer 
```{r bioanalyzer3}
render(here("protocols", "bioanalyzer.Rmd"), params = list(
  num_samples = params$num_seq_lib), run_pandoc = F)
bioanalyzer3 <- bioanalyzer %>% 
  # because this run can't share wells with later runs:
  mutate(quantity = quantity + (12-quantity))
rm(bioanalyzer)
```


### Safe Stopping Point
This is a safe stopping point. If you are stopping, store your sample at ‐15° to ‐25°C.



# Preparation of whole genome libraries 

### Anneal Adapters
```{r anneal2}
render(here("protocols", "anneal.Rmd"), params = list(
    initial_conc_uM = 100,
  p1_adapter_name = "p1_barcoded_adapter_1_ul",
  p2_adapter_name = "p1_barcoded_adapter_2_ul"), run_pandoc = F) 
anneal2 <- anneal %>% 
    mutate(quantity = quantity * params$num_dna_adapters * 5)
rm(anneal)
```
```{r anneal3}
render(here("protocols", "anneal.Rmd"), params = list(
    initial_conc_uM = 100,
  p1_adapter_name = "p2_adapter_1_ul",
  p2_adapter_name = "p2_adapter_2_ul"), run_pandoc = F)
anneal3 <- anneal 
rm(anneal)
```
```{r working2}
# the following makes the working adapters
render(here("protocols", "adapter_working_stock.Rmd"), params = list(
  initial_conc_uM = 40,
  working_conc_uM = 10, 
  final_vol_ul = 100, 
  adpater_stock="p1_stock_adapter_ul"
),run_pandoc = F)

working2 <- working %>% 
  # make for all adapters
  mutate(quantity = quantity * params$num_dna_adapters)
rm(working)
```
```{r working3}
render(here("protocols", "adapter_working_stock.Rmd"), params = list(
  initial_conc_uM = 40,
  working_conc_uM = 10, 
  final_vol_ul = 100, 
  adpater_stock="p2_stock_adapter_ul"
),run_pandoc = F)

working3 <- working 
rm(working)
```

### Extract DNA
```{r extract-dna}
render(here("protocols", "dna_extraction_ali.Rmd"), params = list(
  num_samples = params$num_dna_samples
), run_pandoc = F)

# might need to make liftons
render(here("protocols", "liftons_stock.Rmd"), run_pandoc = F)

# might need to make liftons with enzyme
render(here("protocols", "liftons_w_enzyme_stock.Rmd"), run_pandoc = F)
```

### Fragment DNA

```{r kapa-hyper}
render(here("protocols", "kapa-hyper.Rmd"),params = list(
  num_samples = params$num_dna_samples), run_pandoc = F)
```

### End repair
```{r end-repair}
render(here("protocols", "end-repair.Rmd"),params = list(
  num_samples = params$num_dna_samples), run_pandoc = F)
```


### Adapter ligation
```{r ligate-dna}
 render(here("protocols", "ligation_eecseq.Rmd"), params = list(
  num_samples = params$num_dna_samples), run_pandoc = F)
```


### Post-ligation Cleanup
```{r kapa-clean}
 render(here("protocols", "kapa-clean.Rmd"), params = list(
  sample_vol_ul = 55,
  bead_vol_ul = 44, 
  num_samples = params$num_dna_samples), run_pandoc = F)
```

### Quant samples
```{r quant5}
render(here("protocols", "quant_dna.Rmd"), params = list(
  num_samples = params$num_rna_samples), run_pandoc = F)
  quant5 <- quant
  rm(quant)
```



### Pool samples to be used with in the same index/capture.  
```{r pool}
render(here("protocols", "pool.Rmd"), params = list(
  num_samples_in = params$num_dna_samples, 
  num_pools_out = params$num_seq_lib
))
```


### Library Amplification

```{r amp}
render(here("protocols", "lib-amp_eecseq.Rmd"), params = list(
  num_seq_lib = params$num_seq_lib))
```
```{r working4}
# make working stock of primers
render(here("protocols", "adapter_working_stock.Rmd"), params = list(
  initial_conc_uM = 200,
  working_conc_uM = 10, 
  final_vol_ul = 100, 
  adpater_stock= "f_p1_primer_universal_stock_ul"
),run_pandoc = F)
working4 <- working
rm(working)
```
```{r working5}

render(here("protocols", "adapter_working_stock.Rmd"), params = list(
  initial_conc_uM = 200,
  working_conc_uM = 10, 
  final_vol_ul = 100, 
  adpater_stock= "r_p2_primer_illumina_index_stock_ul"
),run_pandoc = F)

working5 <- working
rm(working)

```


### Post-amplification Cleanup
- Resuspend in 15 ul of 10 mM Tris or water
```{r ampure10}
render(here("protocols", "ampure.Rmd"), params = list(
  num_samples = params$num_seq_lib, 
  sample_vol_ul = 25,
  ampure_vol_ul = 25))
ampure10 <- ampure
rm(ampure)
```


---

### Safe Stopping Point
This is a safe stopping point. If you are stopping, store your sample at ‐15° to ‐25°C.

---
## Hybridization and Capture

### Hybridization

```{r hybrid}

render(here("protocols", "hybridization_eecseq.Rmd"), params = list(
  lib_size_ng = 500, 
  num_seq_lib = 1), run_pandoc = F)

# may need to make 500mM EDTA stock solution - 50mL
render(here("protocols", "edta_stock.Rmd"), params = list(
  final_conc_M = 0.500,
  final_vol_l = 0.050), run_pandoc = F)

# may need to make 10% SDS stock solution - 50mL
render(here("protocols", "sds_stock.Rmd"), params = list(
  final_conc_perc = 10,
  final_vol_ml = 50), run_pandoc = F)

# may need to make TEN stock solution - 50 ml
render(here("protocols", "ten_stock.Rmd"), params = list(
 final_vol_ul = 50000), run_pandoc = F)

# may need to make 20x SSC - 50 ml
render(here("protocols", "ssc_stock.Rmd"), params = list(
  final_vol_l = 0.050), run_pandoc = F)

# may need to make various SSC / 0.1% SDS stock solution - 50 ml
render(here("protocols", "ssc_sds_stock_eecseq.Rmd"), params = list(
 final_vol_ul = 500, 
 perc_ssc = 1), run_pandoc = F)
wash1 <- wash
rm(wash)

render(here("protocols", "ssc_sds_stock_eecseq.Rmd"), params = list(
 final_vol_ul = 500, 
 perc_ssc = 0.5), run_pandoc = F)
wash2 <- wash
rm(wash)

render(here("protocols", "ssc_sds_stock_eecseq.Rmd"), params = list(
 final_vol_ul = 500, 
 perc_ssc = 0.1), run_pandoc = F)
wash3 <- wash
rm(wash)

# make working stocks of blocking oligos
render(here("protocols", "adapter_working_stock.Rmd"), params = list(
 adpater_stock ="blocking_oligo_p5_stock_ul", 
 initial_conc_uM = 200,
  working_conc_uM = 40,
  final_vol_ul = 100), run_pandoc = F)
working <- working %>% 
  # request enough to ping 2 oligos
  mutate(quantity = quantity * 10 * 2)
working8 <- working
rm(working)


render(here("protocols", "adapter_working_stock.Rmd"), params = list(
 adpater_stock ="blocking_oligo_p7_stock_ul", 
 initial_conc_uM = 200,
  working_conc_uM = 40,
  final_vol_ul = 100), run_pandoc = F)
working <- working %>% 
  # request enough to ping 2 oligos
  mutate(quantity = quantity * 10 * 2)
working9 <- working
rm(working)
```
					

### Library re-amplification
```{r reamp}
render(here("protocols", "lib-reamp_eecseq.Rmd"), params = list(
  num_seq_lib = params$num_seq_lib), run_pandoc = F)
```

```{r working6}
# make working stock of primers
render(here("protocols", "adapter_working_stock.Rmd"), params = list(
  initial_conc_uM = 200,
  working_conc_uM = 10, 
  final_vol_ul = 100, 
  adpater_stock="f_p1_primer_universal_stock_ul"
),run_pandoc = F)
working6 <- working
rm(working)

```
```{r working7}
render(here("protocols", "adapter_working_stock.Rmd"), params = list(
  initial_conc_uM = 200,
  working_conc_uM = 10, 
  final_vol_ul = 100, 
  adpater_stock="r_p2_primer_illumina_index_stock_ul"
),run_pandoc = F)

working7 <- working
rm(working)
```

### Perform a 1X SPRI® cleanup by combining the following:
```{r ampure11}
render(here("protocols", "ampure.Rmd"), params = list(
  num_samples = params$num_seq_lib, 
  sample_vol_ul = 25,
  ampure_vol_ul = 25))
ampure11 <- ampure
rm(ampure)

# may need to make 10mM pH 8 Tris-HCl stock solution - 50mL
render(here("protocols", "tris-hcl_stock.Rmd"), params = list(
  final_conc_M = 0.010,
  pH = 8.0,
  final_vol_l = 0.050,
  mol_wt = 121.1))
```


### Quant samples
```{r quant6}
render(here("protocols", "quant_dna.Rmd"), params = list(
  num_samples = params$num_rna_samples), run_pandoc = F)
  quant6 <- quant
  rm(quant)
```

### Verify  
- Run samples on BioAnalyzer/Tape Station/Fragment analyzer  
- check with Nicole Wagner in Debashish's lab for current pricing.  This assumes you are using a High Sensitivity chip (max 7kb per library).
```{r bioanalyzer4}
render(here("protocols", "bioanalyzer.Rmd"), params = list(
  num_samples = params$num_seq_lib), run_pandoc = F)
bioanalyzer4 <- bioanalyzer %>% 
  # because this run can't share wells with later runs:
  mutate(quantity = quantity + (12-quantity))
rm(bioanalyzer)
```
### Sequencing
```{r sequencing}
seq <- tribble(~item, ~quantity,
               "sequencing_lane", params$num_seq_lib)
```


# Combine the lists
```{r combine}
all_lists <- bind_rows(a_tailing, amp, ampure1, ampure2, ampure3, ampure4, ampure5, ampure6, ampure7, ampure8, ampure9, ampure10, ampure11,  anneal1, anneal2, anneal3, annealing, bioanalyzer1, bioanalyzer2, bioanalyzer3, bioanalyzer4, biotin, clean, dsn_norm, dsn_pcr, dyna, edta, enzyme, extr, hybrid, hyper, kapa_clean, liftons, ligation, mrna_capture, nacl, quant5, quant6, pool, quant1, quant2, quant3, quant4, reamp, remove, repair, rna_amp, rna_extr, rna_ligation, sds, synthesis, TEN, tris_hcl, wash3, wash1, wash2, working1, working2, working3, working4, working5, working6, working7, working8, working9,seq)

all_sum <- all_lists %>% 
  group_by(item) %>% 
  summarise(quantity = ceiling(sum(quantity))) %>% 
  # adjust the size of the ampure purchase if needed
  mutate(item = ifelse(item == "ampure_ul", 
                       ifelse(quantity > 60000, "ampure_450", "ampure_60"),item), 
         item = ifelse(item == "ampure_60", 
                       ifelse(quantity < 5000, "ampure_5", "ampure_60"),item),)%>% 
  arrange() %>% 
  # add 10% error 
  mutate(quantity = ceiling(quantity * 1.1))

write_csv(all_sum, here::here("EecSeq", "supply-list.csv"))
```

# Merge with inventory costs
```{r inventory}
# enter code here to look up inventory on google drive
# if these aren't working, restart RStudio, load data from csv, and then run only this line
# all_sum <- read_csv(here::here("EecSeq", "supply-list.csv"))
# run from command line
# inventory_file <- googledrive::drive_get("~/Pinsky Lab Orders/Inventory")
# inventory_file <- drive_get("~/Pinsky Lab Orders/Inventory")
# sheets_deauth()
inventory_raw <- read_sheet(inventory_file, sheet = "Inventory", col_types = "c??ccn???n????") 
inventory <- filter(inventory_raw, Discontinued == "N")

supplies <- left_join(all_sum, inventory, by = "item") %>% 
  select(date, item, quantity, currently_have, vendor, item_number, quantity_sold, price) %>% 
  mutate(cost_per_unit = price/quantity_sold,
         cost_per_project = cost_per_unit * quantity) %>% 
  arrange(price) %>% 
  select(price, vendor, everything())

write_csv(supplies, here::here("EecSeq", "supply-list-with-cost.csv"))
```

### Total project cost 
```{r}
project_cost <- supplies %>% 
  summarise(cost = sum(price, na.rm = T))

cost_per_sample <- project_cost %>% 
  mutate(cost = cost/(params$num_rna_samples + params$num_dna_samples))
```

### What do we really need to purchase
```{r}
to_buy <- supplies %>% 
  filter(quantity >= currently_have, price != 0, vendor != "in house") %>%
  mutate(need = ceiling((quantity - currently_have)/quantity_sold)) %>% 
  select(vendor, need, item, quantity, quantity_sold, currently_have, date, price) %>% 
  arrange(vendor) 

cost_to_buy <- to_buy %>% 
  summarise(cost = sum(price * need, na.rm = T)) 
```

## Method for ordering oligos
*Have to place order through the quick order, which means these don't show up on Pinsky account but they do show up under my account when I log in through quick order portal.*  

I am copying the oligo from Jon's protocol and pasting it into the find window of our master oligo sheet to see if we already have it.  
- SAI1_Adapters - don't expect to have this and didn't find a match.
